<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Comparison, Benchmarking &amp; QC of Epigenomic Datasets ‚Ä¢ EpiCompare</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.7.1/jquery.min.js" integrity="sha512-v2CJ7UaYy4JwqLDIrZUI/4hqeoQieOmAZNXBeQyjo21dadnwR+8ZaIJVT8EE2iyI61OV8e6M8PP2/4hpQINQ/g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="bootstrap-toc.css">
<script src="bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="pkgdown.css" rel="stylesheet">
<script src="pkgdown.js"></script><meta property="og:title" content="Comparison, Benchmarking &amp; QC of Epigenomic Datasets">
<meta property="og:description" content="EpiCompare is used to compare and analyse epigenetic datasets for quality control and benchmarking purposes. The package outputs an HTML report consisting of three sections: (1. General metrics) Metrics on peaks (percentage of blacklisted and non-standard peaks, and peak widths) and fragments (duplication rate) of samples, (2. Peak overlap) Percentage and statistical significance of overlapping and non-overlapping peaks. Also includes upset plot and (3. Functional annotation) functional annotation (ChromHMM, ChIPseeker and enrichment analysis) of peaks. Also includes peak enrichment around TSS.">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">


    <div class="container template-home">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="index.html">EpiCompare</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.13.2</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="articles/EpiCompare.html">Get started</a>
</li>
<li>
  <a href="reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles

    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="articles/docker.html">Docker/Singularity Containers</a>
    </li>
    <li>
      <a href="articles/example_report.html">Example report</a>
    </li>
  </ul>
</li>
<li>
  <a href="news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/neurogenomics/EpiCompare/" class="external-link">
    <span class="fab fa-github fa-lg"></span>

  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->



      </header><div class="row">
  <div class="contents col-md-9">

<div class="section level1">
<div class="page-header"><h1 id="epicompare-qc-and-benchmarking-of-epigenomic-datasets">‚öñ<code>EpiCompare</code>‚öñ<br>QC and Benchmarking of Epigenomic Datasets<a class="anchor" aria-label="anchor" href="#epicompare-qc-and-benchmarking-of-epigenomic-datasets"></a>
</h1></div>
<img src="https://github.com/neurogenomics/EpiCompare/raw/master/inst/hex/hex.png" title="Hex sticker for EpiCompare" height="300"><br><a href="https://www.bioconductor.org/packages/EpiCompare" class="external-link"><img src="https://img.shields.io/badge/release%20version-1.13.1-green.svg"></a> <a href="https://bioconductor.org/packages/stats/bioc/EpiCompare" class="external-link"><img src="https://img.shields.io/badge/download-152/month-green.svg"></a> <a href="https://bioconductor.org/packages/stats/bioc/EpiCompare" class="external-link"><img src="https://img.shields.io/badge/download-2874/total-green.svg"></a> <a href="https://bioconductor.org/packages/stats/bioc/EpiCompare" class="external-link"><img src="http://www.bioconductor.org/shields/downloads/release/EpiCompare.svg" alt="download"></a> <a href="https://cran.r-project.org/web/licenses/GPL-3" class="external-link"><img src="https://img.shields.io/badge/license-GPL--3-blue.svg" alt="License: GPL-3"></a> <a href="https://doi.org/https://doi.org/10.1101/2022.07.22.501149" class="external-link"><img src="https://img.shields.io/badge/doi-https://doi.org/10.1101/2022.07.22.501149-blue.svg"></a> <br><a href="https://github.com/neurogenomics/EpiCompare" class="external-link"><img src="https://img.shields.io/badge/devel%20version-1.13.1-black.svg"></a> <a href="https://github.com/neurogenomics/EpiCompare" class="external-link"><img src="https://img.shields.io/github/languages/code-size/neurogenomics/EpiCompare.svg"></a> <a href="https://github.com/neurogenomics/EpiCompare/commits/master" class="external-link"><img src="https://img.shields.io/github/last-commit/neurogenomics/EpiCompare.svg"></a> <br><a href="https://github.com/neurogenomics/EpiCompare/actions" class="external-link"><img src="https://github.com/neurogenomics/EpiCompare/workflows/rworkflows/badge.svg" alt="R build status"></a> <a href="https://app.codecov.io/gh/neurogenomics/EpiCompare" class="external-link"><img src="https://codecov.io/gh/neurogenomics/EpiCompare/branch/master/graph/badge.svg"></a> <br><a href="https://app.codecov.io/gh/neurogenomics/EpiCompare/tree/master" target="_blank" class="external-link"><img src="https://codecov.io/gh/neurogenomics/EpiCompare/branch/master/graphs/icicle.svg" title="Codecov icicle graph" width="200" height="50" style="vertical-align: top;"></a><br><h4 id="epicompare-qc-and-benchmarking-of-epigenomic-datasets-1">
Authors: <i>Sera Choi, Brian Schilder, Leyla Abbasova, Alan Murphy, Nathan Skene, Thomas Roberts, Hiranyamaya Dash</i>
<a class="anchor" aria-label="anchor" href="#epicompare-qc-and-benchmarking-of-epigenomic-datasets-1"></a>
</h4>
<h5 id="epicompare-qc-and-benchmarking-of-epigenomic-datasets-2">
<i>Updated</i>: Sep-23-2025
<a class="anchor" aria-label="anchor" href="#epicompare-qc-and-benchmarking-of-epigenomic-datasets-2"></a>
</h5>
</div>
<div class="section level1">
<h1 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h1>
<p><code>EpiCompare</code> is an R package for comparing multiple epigenomic datasets for quality control and benchmarking purposes. The function outputs a report in HTML format consisting of three sections:</p>
<ol style="list-style-type: decimal">
<li>
<strong>General Metrics</strong>: Metrics on peaks (percentage of blacklisted and non-standard peaks, and peak widths) and fragments (duplication rate) of samples.</li>
<li>
<strong>Peak Overlap</strong>: Frequency, percentage, statistical significance of overlapping and non-overlapping peaks. This also includes Upset, precision-recall and correlation plots.</li>
<li>
<strong>Functional Annotation</strong>: Functional annotation (ChromHMM, ChIPseeker and enrichment analysis) of peaks. Also includes peak enrichment around Transcription Start Site.</li>
</ol>
<p><em>Note</em>: Peaks located in blacklisted regions and non-standard chromosomes are removed from the files prior to analysis.</p>
</div>
<div class="section level1">
<h1 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h1>
<div class="section level2">
<h2 id="standard">Standard<a class="anchor" aria-label="anchor" href="#standard"></a>
</h2>
<p>To install <code>EpiCompare</code> use:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">require</a></span><span class="op">(</span><span class="st"><a href="https://bioconductor.github.io/BiocManager/" class="external-link">"BiocManager"</a></span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span> <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span></span>
<span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://bioconductor.github.io/BiocManager/reference/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"EpiCompare"</span><span class="op">)</span> </span></code></pre></div>
</div>
<div class="section level2">
<h2 id="all-dependencies">All dependencies<a class="anchor" aria-label="anchor" href="#all-dependencies"></a>
</h2>
<details><summary>
üëà <strong>Details</strong>
</summary><p>Installing all <em>Imports</em> and <em>Suggests</em> will allow you to use the full functionality of <code>EpiCompare</code> right away, without having to stop and install extra dependencies later on.</p>
<p>To install these packages as well, use:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://bioconductor.github.io/BiocManager/reference/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"EpiCompare"</span>, dependencies<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span> </span></code></pre></div>
<p>Note that this will increase installation time, but it means that you won‚Äôt have to worry about installing any R packages when using functions with certain suggested dependencies</p>
</details>
</div>
<div class="section level2">
<h2 id="development">Development<a class="anchor" aria-label="anchor" href="#development"></a>
</h2>
<details><summary>
üëà <strong>Details</strong>
</summary><p>To install the development version of <code>EpiCompare</code>, use:</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">require</a></span><span class="op">(</span><span class="st"><a href="https://remotes.r-lib.org" class="external-link">"remotes"</a></span><span class="op">)</span><span class="op">)</span> <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"remotes"</span><span class="op">)</span></span>
<span><span class="fu">remotes</span><span class="fu">::</span><span class="fu"><a href="https://remotes.r-lib.org/reference/install_github.html" class="external-link">install_github</a></span><span class="op">(</span><span class="st">"neurogenomics/EpiCompare"</span><span class="op">)</span></span></code></pre></div>
</details>
</div>
<div class="section level2">
<h2 id="citation">Citation<a class="anchor" aria-label="anchor" href="#citation"></a>
</h2>
<p>If you use <code>EpiCompare</code>, please cite:</p>
<!-- Modify this by editing the file: inst/CITATION  -->
<blockquote>
<p>EpiCompare: R package for the comparison and quality control of epigenomic peak files (2022) Sera Choi, Brian M. Schilder, Leyla Abbasova, Alan E. Murphy, Nathan G. Skene, bioRxiv, 2022.07.22.501149; doi: <a href="https://doi.org/10.1101/2022.07.22.501149" class="external-link uri">https://doi.org/10.1101/2022.07.22.501149</a></p>
</blockquote>
</div>
</div>
<div class="section level1">
<h1 id="documentation">Documentation<a class="anchor" aria-label="anchor" href="#documentation"></a>
</h1>
<div class="section level2">
<h2 id="epicompare-website">
<a href="https://neurogenomics.github.io/EpiCompare" class="external-link">EpiCompare website</a><a class="anchor" aria-label="anchor" href="#epicompare-website"></a>
</h2>
</div>
<div class="section level2">
<h2 id="dockersingularity-container">
<a href="https://neurogenomics.github.io/EpiCompare/articles/docker" class="external-link">Docker/Singularity container</a><a class="anchor" aria-label="anchor" href="#dockersingularity-container"></a>
</h2>
</div>
<div class="section level2">
<h2 id="bioconductor-page">
<a href="https://doi.org/doi:10.18129/B9.bioc.EpiCompare" class="external-link">Bioconductor page</a><a class="anchor" aria-label="anchor" href="#bioconductor-page"></a>
</h2>
<div class="section level3">
<h3 id="warning-note-on-documentation-versioning">
<span class="emoji" data-emoji="warning">‚ö†Ô∏è</span> Note on documentation versioning<a class="anchor" aria-label="anchor" href="#warning-note-on-documentation-versioning"></a>
</h3>
<p>The documentation in this README and the <a href="https://neurogenomics.github.io/EpiCompare/" class="external-link">GitHub Pages website</a> pertains to the <em>development</em> version of <code>EpiCompare</code>. Older versions of <code>EpiCompare</code> may have slightly different documentation (e.g.¬†available functions, parameters). For documentation in older versions of <code>EpiCompare</code>, please see the <strong>Documentation</strong> section of the relevant version on <a href="https://doi.org/doi:10.18129/B9.bioc.EpiCompare" class="external-link">Bioconductor</a></p>
</div>
</div>
</div>
<div class="section level1">
<h1 id="usage">Usage<a class="anchor" aria-label="anchor" href="#usage"></a>
</h1>
<p>Load package and example datasets.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/neurogenomics/EpiCompare" class="external-link">EpiCompare</a></span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"encode_H3K27ac"</span><span class="op">)</span> <span class="co"># example peakfile</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"CnT_H3K27ac"</span><span class="op">)</span> <span class="co"># example peakfile</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"CnR_H3K27ac"</span><span class="op">)</span> <span class="co"># example peakfile</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"CnT_H3K27ac_picard"</span><span class="op">)</span> <span class="co"># example Picard summary output</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"CnR_H3K27ac_picard"</span><span class="op">)</span> <span class="co"># example Picard summary output</span></span></code></pre></div>
<p>Prepare input files:</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># create named list of peakfiles </span></span>
<span><span class="va">peakfiles</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="st">"CnT"</span><span class="op">=</span><span class="va">CnT_H3K27ac</span>, </span>
<span>                  <span class="st">"CnR"</span><span class="op">=</span><span class="va">CnR_H3K27ac</span><span class="op">)</span> </span>
<span><span class="co"># set ref file and name </span></span>
<span><span class="va">reference</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="st">"ENCODE_H3K27ac"</span> <span class="op">=</span> <span class="va">encode_H3K27ac</span><span class="op">)</span> </span>
<span><span class="co"># create named list of Picard summary</span></span>
<span><span class="va">picard_files</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="st">"CnT"</span><span class="op">=</span><span class="va">CnT_H3K27ac_picard</span>, </span>
<span>                     <span class="st">"CnR"</span><span class="op">=</span><span class="va">CnR_H3K27ac_picard</span><span class="op">)</span> </span></code></pre></div>
<details><summary><strong>üëà Tips on importing user-supplied files</strong>
</summary><p><code><a href="reference/gather_files.html">EpiCompare::gather_files</a></code> is helpful for identifying and importing peak or picard files.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># To import BED files as GRanges object</span></span>
<span><span class="va">peakfiles</span> <span class="op">&lt;-</span> <span class="fu">EpiCompare</span><span class="fu">::</span><span class="fu"><a href="reference/gather_files.html">gather_files</a></span><span class="op">(</span>dir <span class="op">=</span> <span class="st">"path/to/peaks/"</span>, </span>
<span>                                      type <span class="op">=</span> <span class="st">"peaks.stringent"</span><span class="op">)</span></span>
<span><span class="co"># EpiCompare alternatively accepts paths (to BED files) as input </span></span>
<span><span class="va">peakfiles</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>sample1<span class="op">=</span><span class="st">"/path/to/peaks/file1_peaks.stringent.bed"</span>, </span>
<span>                  sample2<span class="op">=</span><span class="st">"/path/to/peaks/file2_peaks.stringent.bed"</span><span class="op">)</span></span>
<span><span class="co"># To import Picard summary output txt file as data frame</span></span>
<span><span class="va">picard_files</span> <span class="op">&lt;-</span> <span class="fu">EpiCompare</span><span class="fu">::</span><span class="fu"><a href="reference/gather_files.html">gather_files</a></span><span class="op">(</span>dir <span class="op">=</span> <span class="st">"path/to/peaks"</span>, </span>
<span>                                         type <span class="op">=</span> <span class="st">"picard"</span><span class="op">)</span></span></code></pre></div>
</details><p>Run <code><a href="reference/EpiCompare.html">EpiCompare()</a></code>:</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" tabindex="-1"></a>EpiCompare<span class="sc">::</span><span class="fu">EpiCompare</span>(<span class="at">peakfiles =</span> peakfiles,</span>
<span id="cb7-2"><a href="#cb7-2" tabindex="-1"></a>                       <span class="at">genome_build =</span> <span class="fu">list</span>(<span class="at">peakfiles=</span><span class="st">"hg19"</span>,</span>
<span id="cb7-3"><a href="#cb7-3" tabindex="-1"></a>                                           <span class="at">reference=</span><span class="st">"hg38"</span>),</span>
<span id="cb7-4"><a href="#cb7-4" tabindex="-1"></a>                       <span class="at">genome_build_output =</span> <span class="st">"hg19"</span>, </span>
<span id="cb7-5"><a href="#cb7-5" tabindex="-1"></a>                       <span class="at">picard_files =</span> picard_files,</span>
<span id="cb7-6"><a href="#cb7-6" tabindex="-1"></a>                       <span class="at">reference =</span> reference,</span>
<span id="cb7-7"><a href="#cb7-7" tabindex="-1"></a>                       <span class="at">run_all =</span> <span class="cn">TRUE</span></span>
<span id="cb7-8"><a href="#cb7-8" tabindex="-1"></a>                       <span class="at">output_dir =</span> <span class="fu">tempdir</span>())</span></code></pre></div>
<div class="section level4">
<h4 id="required-inputs">Required Inputs<a class="anchor" aria-label="anchor" href="#required-inputs"></a>
</h4>
<p>These input parameters must be provided:</p>
<details><summary>
üëà <strong>Details</strong>
</summary><ul>
<li>
<code>peakfiles</code> : Peakfiles you want to analyse. EpiCompare accepts peakfiles as GRanges object and/or as paths to BED files. Files must be listed and named using <code><a href="https://rdrr.io/r/base/list.html" class="external-link">list()</a></code>. E.g. <code>list("name1"=peakfile1, "name2"=peakfile2)</code>.</li>
<li>
<code>genome_build</code> : A named list indicating the human genome build used to generate each of the following inputs:
<ul>
<li>
<code>peakfiles</code> : Genome build for the <code>peakfiles</code> input. Assumes genome build is the same for each element in the <code>peakfiles</code> list.</li>
<li>
<code>reference</code> : Genome build for the <code>reference</code> input.</li>
<li>
<code>blacklist</code> : Genome build for the <code>blacklist</code> input. <br> E.g. <code>genome_build = list(peakfiles="hg38", reference="hg19", blacklist="hg19")</code>
</li>
</ul>
</li>
<li>
<code>genome_build_output</code> Genome build to standardise all inputs to. Liftovers will be performed automatically as needed. Default is ‚Äúhg19‚Äù.</li>
<li>
<code>blacklist</code> : Peakfile as GRanges object specifying genomic regions that have anomalous and/or unstructured signals independent of the cell-line or experiment. For human hg19 and hg38 genome, use built-in data <code>data(hg19_blacklist)</code> and <code>data(hg38_blacklist)</code> respectively. For mouse mm10 genome, use built-in data <code>data(mm10_blacklist)</code>.</li>
<li>
<code>output_dir</code> : Please specify the path to directory, where all <code>EpiCompare</code> outputs will be saved.</li>
</ul></details>
</div>
<div class="section level4">
<h4 id="optional-inputs">Optional Inputs<a class="anchor" aria-label="anchor" href="#optional-inputs"></a>
</h4>
<p>The following input files are optional:</p>
<details><summary>
üëà <strong>Details</strong>
</summary><ul>
<li>
<code>picard_files</code> : A list of summary metrics output from <a href="https://broadinstitute.github.io/picard/" class="external-link">Picard</a>. <em>Picard MarkDuplicates</em> can be used to identify the duplicate reads amongst the alignment. This tool generates a summary output, normally with the ending <em>.markdup.MarkDuplicates.metrics.txt</em>. If this input is provided, metrics on fragments (e.g.¬†mapped fragments and duplication rate) will be included in the report. Files must be in data.frame format and listed using <code><a href="https://rdrr.io/r/base/list.html" class="external-link">list()</a></code> and named using <code><a href="https://rdrr.io/r/base/names.html" class="external-link">names()</a></code>. To import Picard duplication metrics (.txt file) into R as data frame, use <code>picard &lt;- read.table("/path/to/picard/output", header = TRUE, fill = TRUE)</code>.</li>
<li>
<code>reference</code> : Reference peak file(s) is used in <code>stat_plot</code> and <code>chromHMM_plot</code>. File must be in <code>GRanges</code> object, listed and named using <code>list("reference_name" = GRanges_obect)</code>. If more than one reference is specified, <code>EpiCompare</code> outputs individual reports for each reference. However, please note that this can take awhile.</li>
</ul></details>
</div>
<div class="section level4">
<h4 id="optional-plots">Optional Plots<a class="anchor" aria-label="anchor" href="#optional-plots"></a>
</h4>
<p>By default, these plots will not be included in the report unless set to <code>TRUE</code>. To turn on all features at once, simply use the <code>run_all=TRUE</code> argument:</p>
<details><summary>
üëà <strong>Details</strong>
</summary><ul>
<li>
<code>upset_plot</code> : Upset plot of overlapping peaks between samples.</li>
<li>
<code>stat_plot</code> : included only if a <code>reference</code> dataset is provided. The plot shows statistical significance (p/q-values) of sample peaks that are overlapping/non-overlapping with the <code>reference</code> dataset.</li>
<li>
<code>chromHMM_plot</code> : ChromHMM annotation of peaks. If a <code>reference</code> dataset is provided, ChromHMM annotation of overlapping and non-overlapping peaks with the <code>reference</code> is also included in the report.</li>
<li>
<code>chipseeker_plot</code> : ChIPseeker annotation of peaks.</li>
<li>
<code>enrichment_plot</code> : KEGG pathway and GO enrichment analysis of peaks.</li>
<li>
<code>tss_plot</code> : Peak frequency around (+/- 3000bp) transcriptional start site. Note that it may take awhile to generate this plot for large sample sizes.</li>
<li>
<code>precision_recall_plot</code> : Plot showing the precision-recall score across the peak calling stringency thresholds.</li>
<li>
<code>corr_plot</code> : Plot showing the correlation between the quantiles when the genome is binned at a set size. These quantiles are based on the intensity of the peak, dependent on the peak caller used (q-value for MACS2).</li>
</ul></details>
</div>
<div class="section level4">
<h4 id="other-options">Other Options<a class="anchor" aria-label="anchor" href="#other-options"></a>
</h4>
<details><summary>
üëà <strong>Details</strong>
</summary><ul>
<li>
<code>chromHMM_annotation</code> : Cell-line annotation for ChromHMM. Default is K562. Options are:
<ul>
<li>‚ÄúK562‚Äù = K-562 cells</li>
<li>‚ÄúGm12878‚Äù = Cellosaurus cell-line GM12878</li>
<li>‚ÄúH1hesc‚Äù = H1 Human Embryonic Stem Cell</li>
<li>‚ÄúHepg2‚Äù = Hep G2 cell</li>
<li>‚ÄúHmec‚Äù = Human Mammary Epithelial Cell</li>
<li>‚ÄúHsmm‚Äù = Human Skeletal Muscle Myoblasts</li>
<li>‚ÄúHuvec‚Äù = Human Umbilical Vein Endothelial Cells</li>
<li>‚ÄúNhek‚Äù = Normal Human Epidermal Keratinocytes</li>
<li>‚ÄúNhlf‚Äù = Normal Human Lung Fibroblasts</li>
</ul>
</li>
<li>
<code>interact</code> : By default, all heatmaps (percentage overlap and ChromHMM heatmaps) in the report will be interactive. If set FALSE, all heatmaps will be static. N.B. If <code>interact=TRUE</code>, interactive heatmaps will be saved as html files, which may take time for larger sample sizes.</li>
<li>
<code>output_filename</code> : By default, the report is named <em>EpiCompare.html</em>. You can specify the file name of the report here.</li>
<li>
<code>output_timestamp</code> : By default FALSE. If TRUE, the filename of the report includes the date.</li>
</ul></details>
</div>
<div class="section level4">
<h4 id="outputs">Outputs<a class="anchor" aria-label="anchor" href="#outputs"></a>
</h4>
<p><code>EpiCompare</code> outputs the following:</p>
<ol style="list-style-type: decimal">
<li>
<strong>HTML report</strong>: A summary of all analyses saved in specified <code>output_dir</code>
</li>
<li>
<strong>EpiCompare_file</strong>: if <code>save_output=TRUE</code>, all plots generated by <code>EpiCompare</code> will be saved in <em>EpiCompare_file</em> directory also in specified <code>output_dir</code>
</li>
</ol>
<p>An example report comparing ATAC-seq and DNase-seq can be found <a href="https://neurogenomics.github.io/EpiCompare/articles/example_report" class="external-link">here</a></p>
</div>
<div class="section level2">
<h2 id="datasets">Datasets<a class="anchor" aria-label="anchor" href="#datasets"></a>
</h2>
<p><code>EpiCompare</code> includes several built-in datasets:</p>
<details><summary>
üëà <strong>Details</strong>
</summary><ul>
<li>
<code>encode_H3K27ac</code>: Human H3K27ac peak file generated with ChIP-seq using K562 cell-line. Taken from <a href="https://www.encodeproject.org/files/ENCFF044JNJ/" class="external-link">ENCODE</a> project. For more information, run <code><a href="reference/encode_H3K27ac.html">?encode_H3K27ac</a></code>.<br>
</li>
<li>
<code>CnT_H3K27ac</code>: Human H3K27ac peak file generated with CUT&amp;Tag using K562 cell-line from <a href="https://trace.ncbi.nlm.nih.gov/Traces/sra/?run=SRR8383507" class="external-link">Kaya-Okur et al., (2019)</a>. For more information, run <code><a href="reference/CnT_H3K27ac.html">?CnT_H3K27ac</a></code>.</li>
<li>
<code>CnR_H3K27ac</code>: Human H3K27ac peak file generated with CUT&amp;Run using K562 cell-line from <a href="https://trace.ncbi.nlm.nih.gov/Traces/sra/?run=SRR8581604" class="external-link">Meers et al., (2019)</a>. For more details, run <code><a href="reference/CnR_H3K27ac.html">?CnR_H3K27ac</a></code>.</li>
</ul></details>
</div>
<div class="section level2">
<h2 id="contact">Contact<a class="anchor" aria-label="anchor" href="#contact"></a>
</h2>
<div class="section level3">
<h3 id="neurogenomics-lab">
<a href="https://www.neurogenomics.co.uk/inst/report/EpiCompare.html" class="external-link">Neurogenomics Lab</a><a class="anchor" aria-label="anchor" href="#neurogenomics-lab"></a>
</h3>
<p>UK Dementia Research Institute<br>
Department of Brain Sciences<br>
Faculty of Medicine<br>
Imperial College London<br><a href="https://github.com/neurogenomics" class="external-link">GitHub</a><br><a href="https://hub.docker.com/orgs/neurogenomicslab" class="external-link">DockerHub</a></p>
</div>
</div>
<div class="section level2">
<h2 id="session-info">Session Info<a class="anchor" aria-label="anchor" href="#session-info"></a>
</h2>
<details><summary>
üëà <strong>Details</strong>
</summary><div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">utils</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb9"><pre class="sourceCode R"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" tabindex="-1"></a><span class="do">## R version 4.5.1 (2025-06-13)</span></span>
<span id="cb9-2"><a href="#cb9-2" tabindex="-1"></a><span class="do">## Platform: x86_64-pc-linux-gnu</span></span>
<span id="cb9-3"><a href="#cb9-3" tabindex="-1"></a><span class="do">## Running under: Ubuntu 24.04.3 LTS</span></span>
<span id="cb9-4"><a href="#cb9-4" tabindex="-1"></a><span class="do">## </span></span>
<span id="cb9-5"><a href="#cb9-5" tabindex="-1"></a><span class="do">## Matrix products: default</span></span>
<span id="cb9-6"><a href="#cb9-6" tabindex="-1"></a><span class="do">## BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3 </span></span>
<span id="cb9-7"><a href="#cb9-7" tabindex="-1"></a><span class="do">## LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.26.so;  LAPACK version 3.12.0</span></span>
<span id="cb9-8"><a href="#cb9-8" tabindex="-1"></a><span class="do">## </span></span>
<span id="cb9-9"><a href="#cb9-9" tabindex="-1"></a><span class="do">## locale:</span></span>
<span id="cb9-10"><a href="#cb9-10" tabindex="-1"></a><span class="do">##  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              </span></span>
<span id="cb9-11"><a href="#cb9-11" tabindex="-1"></a><span class="do">##  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    </span></span>
<span id="cb9-12"><a href="#cb9-12" tabindex="-1"></a><span class="do">##  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   </span></span>
<span id="cb9-13"><a href="#cb9-13" tabindex="-1"></a><span class="do">##  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 </span></span>
<span id="cb9-14"><a href="#cb9-14" tabindex="-1"></a><span class="do">##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            </span></span>
<span id="cb9-15"><a href="#cb9-15" tabindex="-1"></a><span class="do">## [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       </span></span>
<span id="cb9-16"><a href="#cb9-16" tabindex="-1"></a><span class="do">## </span></span>
<span id="cb9-17"><a href="#cb9-17" tabindex="-1"></a><span class="do">## time zone: UTC</span></span>
<span id="cb9-18"><a href="#cb9-18" tabindex="-1"></a><span class="do">## tzcode source: system (glibc)</span></span>
<span id="cb9-19"><a href="#cb9-19" tabindex="-1"></a><span class="do">## </span></span>
<span id="cb9-20"><a href="#cb9-20" tabindex="-1"></a><span class="do">## attached base packages:</span></span>
<span id="cb9-21"><a href="#cb9-21" tabindex="-1"></a><span class="do">## [1] stats     graphics  grDevices utils     datasets  methods   base     </span></span>
<span id="cb9-22"><a href="#cb9-22" tabindex="-1"></a><span class="do">## </span></span>
<span id="cb9-23"><a href="#cb9-23" tabindex="-1"></a><span class="do">## other attached packages:</span></span>
<span id="cb9-24"><a href="#cb9-24" tabindex="-1"></a><span class="do">## [1] rmarkdown_2.29</span></span>
<span id="cb9-25"><a href="#cb9-25" tabindex="-1"></a><span class="do">## </span></span>
<span id="cb9-26"><a href="#cb9-26" tabindex="-1"></a><span class="do">## loaded via a namespace (and not attached):</span></span>
<span id="cb9-27"><a href="#cb9-27" tabindex="-1"></a><span class="do">##  [1] gtable_0.3.6        jsonlite_2.0.0      renv_1.1.5         </span></span>
<span id="cb9-28"><a href="#cb9-28" tabindex="-1"></a><span class="do">##  [4] dplyr_1.1.4         compiler_4.5.1      BiocManager_1.30.26</span></span>
<span id="cb9-29"><a href="#cb9-29" tabindex="-1"></a><span class="do">##  [7] tidyselect_1.2.1    rvcheck_0.2.1       scales_1.4.0       </span></span>
<span id="cb9-30"><a href="#cb9-30" tabindex="-1"></a><span class="do">## [10] yaml_2.3.10         fastmap_1.2.0       here_1.0.2         </span></span>
<span id="cb9-31"><a href="#cb9-31" tabindex="-1"></a><span class="do">## [13] ggplot2_4.0.0       R6_2.6.1            generics_0.1.4     </span></span>
<span id="cb9-32"><a href="#cb9-32" tabindex="-1"></a><span class="do">## [16] knitr_1.50          yulab.utils_0.2.1   tibble_3.3.0       </span></span>
<span id="cb9-33"><a href="#cb9-33" tabindex="-1"></a><span class="do">## [19] desc_1.4.3          dlstats_0.1.7       rprojroot_2.1.1    </span></span>
<span id="cb9-34"><a href="#cb9-34" tabindex="-1"></a><span class="do">## [22] pillar_1.11.1       RColorBrewer_1.1-3  rlang_1.1.6        </span></span>
<span id="cb9-35"><a href="#cb9-35" tabindex="-1"></a><span class="do">## [25] badger_0.2.5        xfun_0.53           fs_1.6.6           </span></span>
<span id="cb9-36"><a href="#cb9-36" tabindex="-1"></a><span class="do">## [28] S7_0.2.0            cli_3.6.5           magrittr_2.0.4     </span></span>
<span id="cb9-37"><a href="#cb9-37" tabindex="-1"></a><span class="do">## [31] rworkflows_1.0.6    digest_0.6.37       grid_4.5.1         </span></span>
<span id="cb9-38"><a href="#cb9-38" tabindex="-1"></a><span class="do">## [34] rappdirs_0.3.3      lifecycle_1.0.4     vctrs_0.6.5        </span></span>
<span id="cb9-39"><a href="#cb9-39" tabindex="-1"></a><span class="do">## [37] evaluate_1.0.5      glue_1.8.0          data.table_1.17.8  </span></span>
<span id="cb9-40"><a href="#cb9-40" tabindex="-1"></a><span class="do">## [40] farver_2.1.2        tools_4.5.1         pkgconfig_2.0.3    </span></span>
<span id="cb9-41"><a href="#cb9-41" tabindex="-1"></a><span class="do">## [43] htmltools_0.5.8.1</span></span></code></pre></div>
</details><hr>
</div>
</div>

  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <div class="links">
<h2 data-toc-skip>Links</h2>
<ul class="list-unstyled">
<li><a href="https://www.bioconductor.org/packages/EpiCompare" class="external-link">View on Bioconductor</a></li>
<li><a href="https://github.com/neurogenomics/EpiCompare/" class="external-link">Browse source code</a></li>
<li><a href="https://github.com/neurogenomics/EpiCompare/issues" class="external-link">Report a bug</a></li>
</ul>
</div>

<div class="license">
<h2 data-toc-skip>License</h2>
<ul class="list-unstyled">
<li><a href="https://www.r-project.org/Licenses/GPL-3" class="external-link">GPL-3</a></li>
</ul>
</div>


<div class="citation">
<h2 data-toc-skip>Citation</h2>
<ul class="list-unstyled">
<li><a href="authors.html#citation">Citing EpiCompare</a></li>
</ul>
</div>

<div class="developers">
<h2 data-toc-skip>Developers</h2>
<ul class="list-unstyled">
<li>Sera Choi <br><small class="roles"> Author </small> <a href="https://orcid.org/0000-0002-5077-1984" target="orcid.widget" aria-label="ORCID" class="external-link"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a>  </li>
<li>Brian Schilder <br><small class="roles"> Author </small> <a href="https://orcid.org/0000-0001-5949-2191" target="orcid.widget" aria-label="ORCID" class="external-link"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a>  </li>
<li>Leyla Abbasova <br><small class="roles"> Author </small>   </li>
<li>Alan Murphy <br><small class="roles"> Author </small> <a href="https://orcid.org/0000-0002-2487-8753" target="orcid.widget" aria-label="ORCID" class="external-link"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a>  </li>
<li>Nathan Skene <br><small class="roles"> Author </small> <a href="https://orcid.org/0000-0002-6807-3180" target="orcid.widget" aria-label="ORCID" class="external-link"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a>  </li>
<li>Hiranyamaya Dash <br><small class="roles"> Maintainer </small> <a href="https://orcid.org/0009-0005-5514-505X" target="orcid.widget" aria-label="ORCID" class="external-link"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a>  </li>
<li><a href="authors.html">More about authors...</a></li>
</ul>
</div>



  </div>
</div>


      <footer><div class="copyright">
  <p></p>
<p>Developed by Sera Choi, Brian Schilder, Leyla Abbasova, Alan Murphy, Nathan Skene, Hiranyamaya Dash.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.3.</p>
</div>

      </footer>
</div>






  </body>
</html>
