<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Comparison, Benchmarking &amp; QC of Epigenetic Datasets • EpiCompare</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="bootstrap-toc.css">
<script src="bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="pkgdown.css" rel="stylesheet">
<script src="pkgdown.js"></script><meta property="og:title" content="Comparison, Benchmarking &amp; QC of Epigenetic Datasets">
<meta property="og:description" content="EpiCompare is used to compare and analyse epigenetic datasets
    for quality control and benchmarking purposes.  The package outputs an
    HTML report consisting of three sections: (1. General metrics) Metrics
    on peaks (percentage of blacklisted and non-standard peaks, and peak
    widths) and fragments (duplication rate) of samples, (2. Peak overlap)
    Percentage and statistical significance of overlapping and
    non-overlapping peaks. Also includes upset plot and (3. Functional
    annotation) functional annotation (ChromHMM, ChIPseeker and enrichment
    analysis) of peaks.  Also includes peak enrichment around TSS.">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-home">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="index.html">EpiCompare</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.99.20</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="articles/EpiCompare.html">Get started</a>
</li>
<li>
  <a href="reference/index.html">Reference</a>
</li>
<li>
  <a href="news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/neurogenomics/EpiCompare/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="contents col-md-9">

<div class="section level1">
<div class="page-header"><h1 id="epicompare-qc-and-benchmarking-of-epigenetic-datasets">
<code>EpiCompare</code>: QC and Benchmarking of Epigenetic Datasets<a class="anchor" aria-label="anchor" href="#epicompare-qc-and-benchmarking-of-epigenetic-datasets"></a>
</h1></div>
<img src="https://github.com/neurogenomics/EpiCompare/raw/master/inst/hex/hex.png" height="300"><br><br><a href="https://github.com/neurogenomics/EpiCompare" class="external-link"><img src="https://img.shields.io/badge/devel%20version-0.99.20-black.svg"></a> <a href="https://www.bioconductor.org/packages/EpiCompare" class="external-link"><img src="https://img.shields.io/badge/release%20version-1.1.0-green.svg"></a> <a href="https://bioconductor.org/checkResults/devel/bioc-LATEST/EpiCompare" class="external-link"><img src="http://www.bioconductor.org/shields/build/devel/bioc/EpiCompare.svg" alt="BioC status"></a> <a href="https://bioconductor.org/packages/devel/bioc/html/EpiCompare.html#archives" class="external-link"><img src="http://www.bioconductor.org/images/shields/availability/all.svg" alt="platforms"></a> <a href="https://doi.org/https://doi.org/doi:10.18129/B9.bioc.EpiCompare" class="external-link"><img src="https://img.shields.io/badge/doi-https://doi.org/doi:10.18129/B9.bioc.EpiCompare-green.svg"></a> <a href="https://bioconductor.org/packages/stats/bioc/EpiCompare" class="external-link"><img src="https://img.shields.io/badge/download-69/total-green.svg"></a> <a href="https://github.com/neurogenomics/EpiCompare/actions" class="external-link"><img src="https://github.com/neurogenomics/EpiCompare/workflows/R-CMD-check-bioc/badge.svg" alt="R build status"></a> <a href="https://github.com/neurogenomics/EpiCompare/commits/master" class="external-link"><img src="https://img.shields.io/github/last-commit/neurogenomics/EpiCompare.svg"></a> <a href="https://app.codecov.io/gh/neurogenomics/EpiCompare" class="external-link"><img src="https://app.codecov.io/gh/neurogenomics/EpiCompare/branch/master/graph/badge.svg"></a> <a href="https://cran.r-project.org/web/licenses/GPL-3" class="external-link"><img src="https://img.shields.io/badge/license-GPL--3-blue.svg" alt="License: GPL-3"></a>
<h4 id="epicompare-qc-and-benchmarking-of-epigenetic-datasets-1">
Authors: <i>Sera Choi, Brian Schilder, Leyla Abbasova, Alan Murphy, Nathan Skene</i>
<a class="anchor" aria-label="anchor" href="#epicompare-qc-and-benchmarking-of-epigenetic-datasets-1"></a>
</h4>
<h5 id="epicompare-qc-and-benchmarking-of-epigenetic-datasets-2">
<i>Updated</i>: Jun-13-2022
<a class="anchor" aria-label="anchor" href="#epicompare-qc-and-benchmarking-of-epigenetic-datasets-2"></a>
</h5>
</div>
<div class="section level1">
<h1 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h1>
<p><code>EpiCompare</code> is an R package for comparing multiple epigenetic datasets for quality control and benchmarking purposes. The function outputs a report in HTML format consisting of three sections:</p>
<ol style="list-style-type: decimal">
<li>General Metrics: Metrics on peaks (percentage of blacklisted and non-standard peaks, and peak widths) and fragments (duplication rate) of samples.</li>
<li>Peak Overlap: Percentage and statistical significance of overlapping and non-overlapping peaks. Also includes an upset plot.</li>
<li>Functional Annotation: Functional annotation (ChromHMM, ChIPseeker and enrichment analysis) of peaks. Also includes peak enrichment around Transcription Start Site.</li>
</ol>
<p><em>Notes</em>:</p>
<ul>
<li>All functional analyses performed by <code>EpiCompare</code> uses annotations for human genome hg19 or hg38. <br>
</li>
<li>Peaks located in blacklisted regions and non-standard chromosomes are removed from the files prior to analysis.</li>
</ul>
</div>
<div class="section level1">
<h1 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h1>
<p>To install <code>EpiCompare</code> use:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">require</a></span><span class="op">(</span><span class="st">"BiocManager"</span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span>
    <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span>

<span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"EpiCompare"</span><span class="op">)</span> </code></pre></div>
</div>
<div class="section level1">
<h1 id="documentation">Documentation<a class="anchor" aria-label="anchor" href="#documentation"></a>
</h1>
<p><a href="https://neurogenomics.github.io/EpiCompare" class="external-link">EpiCompare website</a></p>
</div>
<div class="section level1">
<h1 id="usage">Usage<a class="anchor" aria-label="anchor" href="#usage"></a>
</h1>
<p>Load package and example datasets.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/neurogenomics/EpiCompare" class="external-link">EpiCompare</a></span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"encode_H3K27ac"</span><span class="op">)</span> <span class="co"># example peakfile</span>
<span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"CnT_H3K27ac"</span><span class="op">)</span> <span class="co"># example peakfile</span>
<span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"CnR_H3K27ac"</span><span class="op">)</span> <span class="co"># example peakfile</span>
<span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"hg19_blacklist"</span><span class="op">)</span> <span class="co"># hg19 blacklist </span>
<span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"CnT_H3K27ac_picard"</span><span class="op">)</span> <span class="co"># example Picard summary output</span>
<span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"CnR_H3K27ac_picard"</span><span class="op">)</span> <span class="co"># example Picard summary output</span></code></pre></div>
<p>Prepare input files:</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">peaklist</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="st">"CnT"</span><span class="op">=</span><span class="va">CnT_H3K27ac</span>, <span class="st">"CnR"</span><span class="op">=</span><span class="va">CnR_H3K27ac</span><span class="op">)</span> <span class="co"># create named list of peakfiles </span>
<span class="va">reference_peak</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="st">"ENCODE_H3K27ac"</span> <span class="op">=</span> <span class="va">encode_H3K27ac</span><span class="op">)</span> <span class="co"># set ref file and name </span>
<span class="va">picard</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="st">"CnT"</span><span class="op">=</span><span class="va">CnT_H3K27ac_picard</span>, <span class="st">"CnR"</span><span class="op">=</span><span class="va">CnR_H3K27ac_picard</span><span class="op">)</span> <span class="co"># create named list of Picard summary</span></code></pre></div>
<p>Additional helps on preparing files:</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># To import BED files as GRanges object</span>
<span class="va">peak</span> <span class="op">&lt;-</span>  <span class="fu">ChIPseeker</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ChIPseeker/man/readPeakFile.html" class="external-link">readPeakFile</a></span><span class="op">(</span><span class="st">"/path/to/peak/file.bed"</span>, as <span class="op">=</span> <span class="st">"GRanges"</span><span class="op">)</span>
<span class="co"># EpiCompare also accepts paths (to BED files) as input </span>
<span class="va">peaklist</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="st">"sample1"</span><span class="op">=</span><span class="st">"/path/to/BED/file1.bed"</span>, 
                 <span class="st">"sample2"</span><span class="op">=</span><span class="st">"/path/to/BED/file2.bed"</span><span class="op">)</span>
<span class="co"># To import Picard summary output txt file as data frame </span>
<span class="va">picard</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.table</a></span><span class="op">(</span><span class="st">"/path/to/Picard/summary.txt"</span>, header <span class="op">=</span> <span class="cn">TRUE</span>, fill <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></code></pre></div>
<p>Run <code><a href="reference/EpiCompare.html">EpiCompare()</a></code>:</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="reference/EpiCompare.html">EpiCompare</a></span><span class="op">(</span>peakfiles <span class="op">=</span> <span class="va">peaklist</span>,
           genome_build <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>peakfiles<span class="op">=</span><span class="st">"hg19"</span>,
                               reference<span class="op">=</span><span class="st">"hg38"</span>,
                               blacklist<span class="op">=</span><span class="st">"hg19"</span><span class="op">)</span>,
           genome_build_output <span class="op">=</span> <span class="st">"hg19"</span>,
           blacklist <span class="op">=</span> <span class="va">hg19_blacklist</span>,
           picard_files <span class="op">=</span> <span class="va">picard</span>,
           reference <span class="op">=</span> <span class="va">reference_peak</span>,
           upset_plot <span class="op">=</span> <span class="cn">TRUE</span>,
           stat_plot <span class="op">=</span> <span class="cn">TRUE</span>,
           chromHMM_plot <span class="op">=</span> <span class="cn">TRUE</span>,
           chromHMM_annotation <span class="op">=</span> <span class="st">"K562"</span>,
           chipseeker_plot <span class="op">=</span> <span class="cn">TRUE</span>,
           enrichment_plot <span class="op">=</span> <span class="cn">TRUE</span>,
           tss_plot <span class="op">=</span> <span class="cn">TRUE</span>,
           interact <span class="op">=</span> <span class="cn">TRUE</span>,
           save_output <span class="op">=</span> <span class="cn">TRUE</span>,
           output_dir <span class="op">=</span> <span class="st">"/path/to/output"</span><span class="op">)</span></code></pre></div>
<div class="section level4">
<h4 id="mandatory-inputs">Mandatory Inputs<a class="anchor" aria-label="anchor" href="#mandatory-inputs"></a>
</h4>
<p>These input parameters must be provided:</p>
<ul>
<li>
<code>peakfiles</code> : Peakfiles you want to analyse. EpiCompare accepts peakfiles as GRanges object and/or as paths to BED files. Files must be listed and named using <code><a href="https://rdrr.io/r/base/list.html" class="external-link">list()</a></code>. E.g. <code>list("name1"=peakfile1, "name2"=peakfile2)</code>.</li>
<li>
<code>genome_build</code> : A named list indicating the human genome build used to generate each of the following inputs:
<ul>
<li>
<code>peakfiles</code> : Genome build for the <code>peakfiles</code> input. Assumes genome build is the same for each element in the <code>peakfiles</code> list.</li>
<li>
<code>reference</code> : Genome build for the <code>reference</code> input.</li>
<li>
<code>blacklist</code> : Genome build for the <code>blacklist</code> input. <br> E.g. <code>genome_build = list(peakfiles="hg38", reference="hg19", blacklist="hg19")</code>
</li>
</ul>
</li>
<li>
<code>genome_build_output</code> Genome build to standardise all inputs to. Liftovers will be performed automatically as needed. Default is “hg19”.</li>
<li>
<code>blacklist</code> : Peakfile as GRanges object specifying genomic regions that have anomalous and/or unstructured signals independent of the cell-line or experiment. For human hg19 and hg38 genome, use built-in data <code>data(hg19_blacklist)</code> and <code>data(hg38_blacklist)</code> respectively.</li>
<li>
<code>output_dir</code> : Please specify the path to directory, where all EpiCompare outputs will be saved.</li>
</ul>
</div>
<div class="section level4">
<h4 id="optional-inputs">Optional Inputs<a class="anchor" aria-label="anchor" href="#optional-inputs"></a>
</h4>
<ul>
<li>
<code>picard_files</code> : A list of summary metrics output from <a href="https://broadinstitute.github.io/picard/" class="external-link">Picard</a>. <em>Picard MarkDuplicates</em> can be used to identify the duplicate reads amongst the alignment. This tool generates a summary output, normally with the ending <em>.markdup.MarkDuplicates.metrics.txt</em>. If this input is provided, metrics on fragments (e.g. mapped fragments and duplication rate) will be included in the report. Files must be in data.frame format and listed using <code><a href="https://rdrr.io/r/base/list.html" class="external-link">list()</a></code> and named using <code><a href="https://rdrr.io/r/base/names.html" class="external-link">names()</a></code>. To import Picard duplication metrics (.txt file) into R as data frame, use <code>picard &lt;- read.table("/path/to/picard/output", header = TRUE, fill = TRUE)</code>.</li>
<li>
<code>reference</code> : Reference peak file(s) is used in <code>stat_plot</code> and <code>chromHMM_plot</code>. File must be in GRanges object, listed and named using <code>list("reference_name" = GRanges_obect)</code>. If more than one reference is specified, EpiCompare outputs individual reports for each reference. However, please note that this can take awhile.</li>
</ul>
</div>
<div class="section level4">
<h4 id="optional-plots">Optional Plots<a class="anchor" aria-label="anchor" href="#optional-plots"></a>
</h4>
<p>By default, these plots will not be included in the report unless set <code>TRUE</code>.</p>
<ul>
<li>
<code>upset_plot</code> : Upset plot of overlapping peaks between samples.</li>
<li>
<code>stat_plot</code> : included only if a <code>reference</code> dataset is provided. The plot shows statistical significance (p/q-values) of sample peaks that are overlapping/non-overlapping with the <code>reference</code> dataset.</li>
<li>
<code>chromHMM_plot</code> : ChromHMM annotation of peaks. If a <code>reference</code> dataset is provided, ChromHMM annotation of overlapping and non-overlapping peaks with the <code>reference</code> is also included in the report.</li>
<li>
<code>chipseeker_plot</code> : ChIPseeker annotation of peaks.</li>
<li>
<code>enrichment_plot</code> : KEGG pathway and GO enrichment analysis of peaks.</li>
<li>
<code>tss_plot</code> : Peak frequency around (+/- 3000bp) transcriptional start site. Note that it may take awhile to generate this plot for large sample sizes.</li>
</ul>
</div>
<div class="section level4">
<h4 id="other-options">Other Options<a class="anchor" aria-label="anchor" href="#other-options"></a>
</h4>
<ul>
<li>
<code>chromHMM_annotation</code> : Cell-line annotation for ChromHMM. Default is K562. Options are:
<ul>
<li>“K562” = K-562 cells</li>
<li>“Gm12878” = Cellosaurus cell-line GM12878</li>
<li>“H1hesc” = H1 Human Embryonic Stem Cell</li>
<li>“Hepg2” = Hep G2 cell</li>
<li>“Hmec” = Human Mammary Epithelial Cell</li>
<li>“Hsmm” = Human Skeletal Muscle Myoblasts</li>
<li>“Huvec” = Human Umbilical Vein Endothelial Cells</li>
<li>“Nhek” = Normal Human Epidermal Keratinocytes</li>
<li>“Nhlf” = Normal Human Lung Fibroblasts</li>
</ul>
</li>
<li>
<code>interact</code> : By default, all heatmaps (percentage overlap and ChromHMM heatmaps) in the report will be interactive. If set FALSE, all heatmaps will be static. N.B. If <code>interact=TRUE</code>, interactive heatmaps will be saved as html files, which may take time for larger sample sizes.</li>
<li>
<code>output_filename</code> : By default, the report is named EpiCompare.html. You can specify the filename of the report here.</li>
<li>
<code>output_timestamp</code> : By default FALSE. If TRUE, the filename of the report includes the date.</li>
</ul>
</div>
<div class="section level4">
<h4 id="outputs">Outputs<a class="anchor" aria-label="anchor" href="#outputs"></a>
</h4>
<p><code>EpiCompare</code> outputs the following:</p>
<ol style="list-style-type: decimal">
<li>HTML report: A summary of all analyses saved in specified <code>output_dir</code>
</li>
<li>EpiCompare_file: if <code>save_output=TRUE</code>, all plots generated by EpiCompare will be saved in EpiCompare_file directory also in specified <code>output_dir</code>
</li>
</ol>
</div>
</div>
<div class="section level1">
<h1 id="session-info">Session Info<a class="anchor" aria-label="anchor" href="#session-info"></a>
</h1>
<details><div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">utils</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<div class="sourceCode" id="cb7"><pre class="sourceCode R"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a><span class="do">## R version 4.2.0 (2022-04-22)</span></span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a><span class="do">## Platform: x86_64-pc-linux-gnu (64-bit)</span></span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a><span class="do">## Running under: Ubuntu 20.04.4 LTS</span></span>
<span id="cb7-4"><a href="#cb7-4" aria-hidden="true" tabindex="-1"></a><span class="do">## </span></span>
<span id="cb7-5"><a href="#cb7-5" aria-hidden="true" tabindex="-1"></a><span class="do">## Matrix products: default</span></span>
<span id="cb7-6"><a href="#cb7-6" aria-hidden="true" tabindex="-1"></a><span class="do">## BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3</span></span>
<span id="cb7-7"><a href="#cb7-7" aria-hidden="true" tabindex="-1"></a><span class="do">## LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/liblapack.so.3</span></span>
<span id="cb7-8"><a href="#cb7-8" aria-hidden="true" tabindex="-1"></a><span class="do">## </span></span>
<span id="cb7-9"><a href="#cb7-9" aria-hidden="true" tabindex="-1"></a><span class="do">## locale:</span></span>
<span id="cb7-10"><a href="#cb7-10" aria-hidden="true" tabindex="-1"></a><span class="do">##  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              </span></span>
<span id="cb7-11"><a href="#cb7-11" aria-hidden="true" tabindex="-1"></a><span class="do">##  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    </span></span>
<span id="cb7-12"><a href="#cb7-12" aria-hidden="true" tabindex="-1"></a><span class="do">##  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   </span></span>
<span id="cb7-13"><a href="#cb7-13" aria-hidden="true" tabindex="-1"></a><span class="do">##  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 </span></span>
<span id="cb7-14"><a href="#cb7-14" aria-hidden="true" tabindex="-1"></a><span class="do">##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            </span></span>
<span id="cb7-15"><a href="#cb7-15" aria-hidden="true" tabindex="-1"></a><span class="do">## [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       </span></span>
<span id="cb7-16"><a href="#cb7-16" aria-hidden="true" tabindex="-1"></a><span class="do">## </span></span>
<span id="cb7-17"><a href="#cb7-17" aria-hidden="true" tabindex="-1"></a><span class="do">## attached base packages:</span></span>
<span id="cb7-18"><a href="#cb7-18" aria-hidden="true" tabindex="-1"></a><span class="do">## [1] stats     graphics  grDevices utils     datasets  methods   base     </span></span>
<span id="cb7-19"><a href="#cb7-19" aria-hidden="true" tabindex="-1"></a><span class="do">## </span></span>
<span id="cb7-20"><a href="#cb7-20" aria-hidden="true" tabindex="-1"></a><span class="do">## other attached packages:</span></span>
<span id="cb7-21"><a href="#cb7-21" aria-hidden="true" tabindex="-1"></a><span class="do">## [1] rmarkdown_2.14</span></span>
<span id="cb7-22"><a href="#cb7-22" aria-hidden="true" tabindex="-1"></a><span class="do">## </span></span>
<span id="cb7-23"><a href="#cb7-23" aria-hidden="true" tabindex="-1"></a><span class="do">## loaded via a namespace (and not attached):</span></span>
<span id="cb7-24"><a href="#cb7-24" aria-hidden="true" tabindex="-1"></a><span class="do">##  [1] BiocManager_1.30.18 pillar_1.7.0        compiler_4.2.0     </span></span>
<span id="cb7-25"><a href="#cb7-25" aria-hidden="true" tabindex="-1"></a><span class="do">##  [4] RColorBrewer_1.1-3  yulab.utils_0.0.4   tools_4.2.0        </span></span>
<span id="cb7-26"><a href="#cb7-26" aria-hidden="true" tabindex="-1"></a><span class="do">##  [7] digest_0.6.29       jsonlite_1.8.0      evaluate_0.15      </span></span>
<span id="cb7-27"><a href="#cb7-27" aria-hidden="true" tabindex="-1"></a><span class="do">## [10] lifecycle_1.0.1     tibble_3.1.7        gtable_0.3.0       </span></span>
<span id="cb7-28"><a href="#cb7-28" aria-hidden="true" tabindex="-1"></a><span class="do">## [13] pkgconfig_2.0.3     rlang_1.0.2         cli_3.3.0          </span></span>
<span id="cb7-29"><a href="#cb7-29" aria-hidden="true" tabindex="-1"></a><span class="do">## [16] DBI_1.1.2           rvcheck_0.2.1       yaml_2.3.5         </span></span>
<span id="cb7-30"><a href="#cb7-30" aria-hidden="true" tabindex="-1"></a><span class="do">## [19] xfun_0.31           fastmap_1.1.0       stringr_1.4.0      </span></span>
<span id="cb7-31"><a href="#cb7-31" aria-hidden="true" tabindex="-1"></a><span class="do">## [22] dplyr_1.0.9         knitr_1.39          desc_1.4.1         </span></span>
<span id="cb7-32"><a href="#cb7-32" aria-hidden="true" tabindex="-1"></a><span class="do">## [25] generics_0.1.2      vctrs_0.4.1         dlstats_0.1.5      </span></span>
<span id="cb7-33"><a href="#cb7-33" aria-hidden="true" tabindex="-1"></a><span class="do">## [28] rprojroot_2.0.3     grid_4.2.0          tidyselect_1.1.2   </span></span>
<span id="cb7-34"><a href="#cb7-34" aria-hidden="true" tabindex="-1"></a><span class="do">## [31] glue_1.6.2          R6_2.5.1            fansi_1.0.3        </span></span>
<span id="cb7-35"><a href="#cb7-35" aria-hidden="true" tabindex="-1"></a><span class="do">## [34] ggplot2_3.3.6       purrr_0.3.4         badger_0.2.1       </span></span>
<span id="cb7-36"><a href="#cb7-36" aria-hidden="true" tabindex="-1"></a><span class="do">## [37] magrittr_2.0.3      scales_1.2.0        ellipsis_0.3.2     </span></span>
<span id="cb7-37"><a href="#cb7-37" aria-hidden="true" tabindex="-1"></a><span class="do">## [40] htmltools_0.5.2     assertthat_0.2.1    colorspace_2.0-3   </span></span>
<span id="cb7-38"><a href="#cb7-38" aria-hidden="true" tabindex="-1"></a><span class="do">## [43] utf8_1.2.2          stringi_1.7.6       munsell_0.5.0      </span></span>
<span id="cb7-39"><a href="#cb7-39" aria-hidden="true" tabindex="-1"></a><span class="do">## [46] crayon_1.5.1</span></span></code></pre></div>
</details><div class="section level2">
<h2 id="contact">Contact<a class="anchor" aria-label="anchor" href="#contact"></a>
</h2>
<div class="section level3">
<h3 id="neurogenomics-lab">
<a href="https://www.neurogenomics.co.uk/inst/report/EpiCompare.html" class="external-link">Neurogenomics Lab</a><a class="anchor" aria-label="anchor" href="#neurogenomics-lab"></a>
</h3>
<p>UK Dementia Research Institute<br>
Department of Brain Sciences<br>
Faculty of Medicine<br>
Imperial College London<br><a href="https://github.com/neurogenomics" class="external-link">GitHub</a><br><a href="https://hub.docker.com/orgs/neurogenomicslab" class="external-link">DockerHub</a></p>
<p><br></p>
</div>
</div>
</div>

  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <div class="links">
<h2 data-toc-skip>Links</h2>
<ul class="list-unstyled">
<li><a href="https://www.bioconductor.org/packages/EpiCompare" class="external-link">View on Bioconductor</a></li>
<li><a href="https://github.com/neurogenomics/EpiCompare/" class="external-link">Browse source code</a></li>
<li><a href="https://github.com/neurogenomics/EpiCompare/issues" class="external-link">Report a bug</a></li>
</ul>
</div>

<div class="license">
<h2 data-toc-skip>License</h2>
<ul class="list-unstyled">
<li><a href="https://www.r-project.org/Licenses/GPL-3" class="external-link">GPL-3</a></li>
</ul>
</div>


<div class="citation">
<h2 data-toc-skip>Citation</h2>
<ul class="list-unstyled">
<li><a href="authors.html#citation">Citing EpiCompare</a></li>
</ul>
</div>

<div class="developers">
<h2 data-toc-skip>Developers</h2>
<ul class="list-unstyled">
<li>Sera Choi <br><small class="roles"> Author, maintainer </small> <a href="https://orcid.org/0000-0002-5077-1984" target="orcid.widget" aria-label="ORCID" class="external-link"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
<li>Brian Schilder <br><small class="roles"> Author </small> <a href="https://orcid.org/0000-0001-5949-2191" target="orcid.widget" aria-label="ORCID" class="external-link"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
<li>Leyla Abbasova <br><small class="roles"> Author </small>  </li>
<li>Alan Murphy <br><small class="roles"> Author </small> <a href="https://orcid.org/0000-0002-2487-8753" target="orcid.widget" aria-label="ORCID" class="external-link"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
<li>Nathan Skene <br><small class="roles"> Author </small> <a href="https://orcid.org/0000-0002-6807-3180" target="orcid.widget" aria-label="ORCID" class="external-link"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
</ul>
</div>



  </div>
</div>


      <footer><div class="copyright">
  <p></p>
<p>Developed by Sera Choi, Brian Schilder, Leyla Abbasova, Alan Murphy, Nathan Skene.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.4.9000.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
