<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Compute consensus peaks — compute_consensus_peaks • EpiCompare</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Compute consensus peaks — compute_consensus_peaks"><meta property="og:description" content="Compute consensus peaks from a list of GRanges."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">EpiCompare</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.1.4</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../articles/EpiCompare.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/docker.html">Docker/Singularity Containers</a>
    </li>
  </ul></li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/neurogenomics/EpiCompare/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Compute consensus peaks</h1>
    <small class="dont-index">Source: <a href="https://github.com/neurogenomics/EpiCompare/blob/HEAD/R/compute_consensus_peaks.R" class="external-link"><code>R/compute_consensus_peaks.R</code></a></small>
    <div class="hidden name"><code>compute_consensus_peaks.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Compute consensus peaks from a list of <a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html" class="external-link">GRanges</a>.</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">compute_consensus_peaks</span><span class="op">(</span></span>
<span>  <span class="va">grlist</span>,</span>
<span>  groups <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  <span class="va">genome_build</span>,</span>
<span>  lower <span class="op">=</span> <span class="fl">2</span>,</span>
<span>  upper <span class="op">=</span> <span class="cn">Inf</span>,</span>
<span>  min.gapwidth <span class="op">=</span> <span class="fl">1L</span>,</span>
<span>  method <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"granges"</span>, <span class="st">"consensusseeker"</span><span class="op">)</span>,</span>
<span>  <span class="va">...</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="source">
    <h2>Source</h2>
    <p><a href="https://ro-che.info/articles/2018-07-11-chip-seq-consensus" class="external-link">
GenomicRanges tutorial</a></p>
<p><a href="https://doi.org/doi:10.18129/B9.bioc.consensusSeekeR" class="external-link">
consensusSeekeR</a></p>
    </div>
    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>grlist</dt>
<dd><p>Named list of <a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html" class="external-link">GRanges</a> objects.</p></dd>


<dt>groups</dt>
<dd><p>A character vector of the same length as <code>grlist</code> 
defining how to group <a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html" class="external-link">GRanges</a> objects when 
computing consensus peaks.</p></dd>


<dt>genome_build</dt>
<dd><p>Genome build name.</p></dd>


<dt>lower, upper</dt>
<dd><p>The lower and upper bounds for the slice.</p></dd>


<dt>min.gapwidth</dt>
<dd><p>Ranges separated by a gap of at least <code>min.gapwidth</code> positions
    are not merged.</p></dd>


<dt>method</dt>
<dd><p>Method to call peaks with:</p><ul><li><p>"granges" : Simple overlap procedure using 
<a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html" class="external-link">GRanges</a> functions.
Faster but less accurate.</p></li>
<li><p>"consensusseeker" : 
Uses <a href="https://rdrr.io/pkg/consensusSeekeR/man/findConsensusPeakRegions.html" class="external-link">findConsensusPeakRegions</a> to compute consensus
peaks. 
Slower but more accurate.</p></li>
</ul></dd>


<dt>...</dt>
<dd><p>Arguments passed on to <code><a href="https://rdrr.io/pkg/consensusSeekeR/man/findConsensusPeakRegions.html" class="external-link">consensusSeekeR::findConsensusPeakRegions</a></code></p><dl><dt><code>narrowPeaks</code></dt>
<dd><p>a <code>GRanges</code> containing
called peak regions of signal enrichment based on pooled, normalized data
for all analyzed experiments. All <code>GRanges</code> entries must
have a metadata field called "name" which identifies the region to
the called peak. All <code>GRanges</code> entries must also
have a row name which identifies the experiment of origin. Each
<code>peaks</code> entry must have an associated <code>narrowPeaks</code> entry.
A <code>GRanges</code> entry is associated to a <code>narrowPeaks</code> entry by
having a identical metadata "name" field and a identical row name.</p></dd>

    <dt><code>peaks</code></dt>
<dd><p>a <code>GRanges</code> containing called peaks of signal
enrichment based on pooled, normalized data
for all analyzed experiments. All <code>GRanges</code> entries must
have a metadata field called "name" which identifies the called
peak. All <code>GRanges</code> entries must
have a row name which identifies the experiment of origin. Each
<code>peaks</code> entry must have an associated <code>narrowPeaks</code> entry. A
<code>GRanges</code> entry is associated to a <code>narrowPeaks</code> entry by having
a identical metadata "name" field and a identical row name.</p></dd>

    <dt><code>chrInfo</code></dt>
<dd><p>a <code>Seqinfo</code> containing the name and the length of the
chromosomes to analyze. Only the chomosomes contained in this
<code>Seqinfo</code> will be analyzed.</p></dd>

    <dt><code>extendingSize</code></dt>
<dd><p>a <code>numeric</code> value indicating the size of padding
on both sides of the position of the peaks median to create the
consensus region. The minimum size of the consensus region is
equal to twice the value of the <code>extendingSize</code> parameter.
The size of the <code>extendingSize</code> must be a positive integer.
Default = 250.</p></dd>

    <dt><code>expandToFitPeakRegion</code></dt>
<dd><p>a <code>logical</code> indicating if the region size,
which is set by the <code>extendingSize</code> parameter is extended to include
the entire narrow peak regions of all peaks included in the unextended
consensus region. The narrow peak regions of the peaks added because of the
extension are not considered for the extension. Default: <code>FALSE</code>.</p></dd>

    <dt><code>shrinkToFitPeakRegion</code></dt>
<dd><p>a <code>logical</code> indicating if the region size,
which is set by the <code>extendingSize</code> parameter is shrinked to
fit the narrow peak regions of the peaks when all those regions
are smaller than the consensus region. Default: <code>FALSE</code>.</p></dd>

    <dt><code>minNbrExp</code></dt>
<dd><p>a positive <code>numeric</code> or a positive <code>integer</code>
indicating the minimum number of experiments in which at least one peak
must be present for a potential consensus region. The numeric must be a
positive integer inferior or equal to the number of experiments present
in the <code>narrowPeaks</code> and <code>peaks</code> parameters. Default = 1.</p></dd>

    <dt><code>nbrThreads</code></dt>
<dd><p>a <code>numeric</code> or a <code>integer</code> indicating the
number of threads to use in parallel. The <code>nbrThreads</code> must be a
positive integer. Default = 1.</p></dd>

  
</dl></dd>

</dl></div>
    <div id="value">
    <h2>Value</h2>
    

<p>Named list of consensus peak <a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html" class="external-link">GRanges</a>.</p>
    </div>
    <div id="details">
    <h2>Details</h2>
    <p><em>NOTE:</em> If you get the error 
<code>"Error in serialize(data, node$con) : error writing to connection"</code>,
try running <a href="https://rdrr.io/r/base/showConnections.html" class="external-link">closeAllConnections</a> 
and rerun compute_consensus_peaks. 
This error can sometimes occur when 
compute_consensus_peaks
has been disrupted partway through.</p>
    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"encode_H3K27ac"</span><span class="op">)</span> <span class="co"># example dataset as GRanges object</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"CnT_H3K27ac"</span><span class="op">)</span> <span class="co"># example dataset as GRanges object</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"CnR_H3K27ac"</span><span class="op">)</span> <span class="co"># example dataset as GRanges object </span></span></span>
<span class="r-in"><span><span class="va">grlist</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>CnR<span class="op">=</span><span class="va">CnR_H3K27ac</span>, CnT<span class="op">=</span><span class="va">CnT_H3K27ac</span>, ENCODE<span class="op">=</span><span class="va">encode_H3K27ac</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="va">consensus_peaks</span> <span class="op">&lt;-</span> <span class="fu">compute_consensus_peaks</span><span class="op">(</span>grlist <span class="op">=</span> <span class="va">grlist</span>,</span></span>
<span class="r-in"><span>                                           groups <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Imperial"</span>,</span></span>
<span class="r-in"><span>                                                      <span class="st">"Imperial"</span>,</span></span>
<span class="r-in"><span>                                                      <span class="st">"ENCODE"</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Removing non-standard chromosomes.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Computing consensus peaks for group: Imperial </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - 2 files </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - 1,670-2,707 peaks each</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Identified 1,325 consensus peaks from 2 peak files</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Computing consensus peaks for group: ENCODE </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - 1 files </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - 5,142-5,142 peaks each</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> WARNING: Cannot compute consensus peaks when group has &lt;2 peak files. Returning original GRanges object instead.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Done computing consensus peaks in 0 min.</span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Sera Choi, Brian Schilder, Leyla Abbasova, Alan Murphy, Nathan Skene.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer></div>

  


  

  </body></html>

