<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>`EpiCompare`: Getting started • EpiCompare</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.7.1/jquery.min.js" integrity="sha512-v2CJ7UaYy4JwqLDIrZUI/4hqeoQieOmAZNXBeQyjo21dadnwR+8ZaIJVT8EE2iyI61OV8e6M8PP2/4hpQINQ/g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="`EpiCompare`: Getting started">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">


    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">EpiCompare</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.13.2</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../articles/EpiCompare.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles

    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/docker.html">Docker/Singularity Containers</a>
    </li>
    <li>
      <a href="../articles/example_report.html">Example report</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/neurogenomics/EpiCompare/" class="external-link">
    <span class="fab fa-github fa-lg"></span>

  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->



      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>
<code>EpiCompare</code>: Getting started</h1>
                        <h4 data-toc-skip class="author"><h4>
<br>
Authors: <i>Sera Choi, Brian Schilder, Leyla Abbasova, Alan Murphy,
Nathan Skene, Thomas Roberts, Hiranyamaya Dash</i><br>
</h4></h4>
            
            <h4 data-toc-skip class="date"><h4>
<br>
Vignette updated: <i>Sep-23-2025</i><br>
</h4></h4>
      
      <small class="dont-index">Source: <a href="https://github.com/neurogenomics/EpiCompare/blob/ggplot-fix/vignettes/EpiCompare.Rmd" class="external-link"><code>vignettes/EpiCompare.Rmd</code></a></small>
      <div class="hidden name"><code>EpiCompare.Rmd</code></div>

    </div>

    
    
<div class="section level2">
<h2 id="overview">Overview<a class="anchor" aria-label="anchor" href="#overview"></a>
</h2>
<p>The <em>EpiCompare</em> package is designed to facilitate the
comparison of epigenomic datasets for quality control and benchmarking
purposes. The package combines several downstream analysis tools for
epigenomic data and generates a single report that collates all results
of the analysis. This allows users to conduct downstream analysis of
multiple epigenomic datasets simultaneously and compare the results in a
simple and efficient way.</p>
<div class="section level3">
<h3 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h3>
<p>For many years, ChIP-seq has been the standard method for epigenomic
profiling, but it suffers from a host of limitations. Recently, many
other epigenomic technologies (e.g. CUT&amp;Run, CUT&amp;Tag and TIP-seq
etc.), designed to overcome these constraints, have been developed. To
better understand the performance of these novel approaches, it is
important that we systematically compare these technologies and
benchmark against a “gold-standard”.</p>
<p>There are many tools in R (e.g. <em>ChIPseeker</em>) that can be used
to conduct downstream analysis and comparison of epigenomic datasets.
However, these are often scattered across different packages and
difficult to use for researchers with none or little computational
experience.</p>
<p><em>EpiCompare</em> is designed to provide a simple and comprehensive
way of analysing and comparing epigenomic datasets. It combines many
useful downstream analysis tools, which can easily be controlled by
users and it collates the results in a single report. This allows
researchers to systematically compare different epigenomic
technologies.</p>
<p>While the main functionality of <em>EpiCompare</em> is to contrast
epigenomic technologies, it can also be used to compare datasets
generated using different experimental conditions and data analysis
workflows of one technology. This can help researchers to establish a
consensus regarding the optimal use of the method.</p>
<p>Currently, <em>EpiCompare</em> supports human and mouse genomes
(hg19, hg38, mm9 and mm10).</p>
</div>
</div>
<div class="section level2">
<h2 id="data">Data<a class="anchor" aria-label="anchor" href="#data"></a>
</h2>
<p>The <em>EpiCompare</em> package contains a small subset of histone
mark H3K27ac profile data obtained/generated from:</p>
<ul>
<li>ENCODE (data accession: ENCFF044JNJ)</li>
<li>CUT&amp;Tag from Kaya-Okur et al., (2019). (PMID: 31036827)</li>
<li>CUT&amp;Run from Meers et al., (2019). (PMID: 31232687)</li>
</ul>
<p>It also contains human genome hg19 and hg38 blacklisted regions
obtained from ENCODE. The ENCODE blacklist includes regions of the
genome that have anomalous and/or unstructured signals independent of
the cell-line or experiment. Removal of ENCODE blacklist is recommended
for quality measure.</p>
<p>These dataset will be used to showcase <em>EpiCompare</em>
functionality</p>
</div>
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<p>To install the package, run the following:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">require</a></span><span class="op">(</span><span class="st"><a href="https://bioconductor.github.io/BiocManager/" class="external-link">"BiocManager"</a></span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://bioconductor.github.io/BiocManager/reference/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"EpiCompare"</span><span class="op">)</span> </span></code></pre></div>
</div>
<div class="section level2">
<h2 id="running-epicompare">Running <em>EpiCompare</em><a class="anchor" aria-label="anchor" href="#running-epicompare"></a>
</h2>
<p>In this example analysis, we will compare CUT&amp;Run and CUT&amp;Tag
of histone mark H3K27ac against ENCODE ChIP-seq.</p>
<div class="section level3">
<h3 id="load-package-and-example-datasets">Load package and example datasets<a class="anchor" aria-label="anchor" href="#load-package-and-example-datasets"></a>
</h3>
<p>Once installed, load the package:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/neurogenomics/EpiCompare" class="external-link">EpiCompare</a></span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## </span></span></code></pre>
<pre><code><span><span class="co">## Warning: replacing previous import 'Biostrings::pattern' by 'grid::pattern'</span></span>
<span><span class="co">## when loading 'genomation'</span></span></code></pre>
<p>Load example datasets used in this analysis:</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"encode_H3K27ac"</span><span class="op">)</span> <span class="co"># ENCODE ChIP-seq</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"CnT_H3K27ac"</span><span class="op">)</span> <span class="co"># CUT&amp;Tag</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"CnR_H3K27ac"</span><span class="op">)</span> <span class="co"># CUT&amp;Run</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"hg19_blacklist"</span><span class="op">)</span> <span class="co"># hg19 genome blacklist </span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"CnT_H3K27ac_picard"</span><span class="op">)</span> <span class="co"># CUT&amp;Tag Picard summary output</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"CnR_H3K27ac_picard"</span><span class="op">)</span> <span class="co"># CUT&amp;Run Picard summary output</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="prepare-input-data">Prepare input data<a class="anchor" aria-label="anchor" href="#prepare-input-data"></a>
</h3>
<div class="section level4">
<h4 id="peaklist">Peaklist<a class="anchor" aria-label="anchor" href="#peaklist"></a>
</h4>
<p><em>EpiCompare</em> accepts datasets both as <code>GRanges</code>
object and as paths to BED files. Peakfiles (<code>GRanges</code> or
paths) that you would like to analyse must be listed and named (see
below).</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># To import BED files as GRanges object</span></span>
<span><span class="va">peak_GRanges</span> <span class="op">&lt;-</span><span class="fu">ChIPseeker</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ChIPseeker/man/readPeakFile.html" class="external-link">readPeakFile</a></span><span class="op">(</span><span class="st">"/path/to/peak/file.bed"</span>,as <span class="op">=</span> <span class="st">"GRanges"</span><span class="op">)</span></span>
<span><span class="co"># EpiCompare also accepts paths (to BED files) as input </span></span>
<span><span class="va">peak_path</span> <span class="op">&lt;-</span> <span class="st">"/path/to/BED/file1.bed"</span></span>
<span><span class="co"># Create named peak list</span></span>
<span><span class="va">peaklist</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="va">peak_GRanges</span>, <span class="va">peak_path</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">peaklist</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"sample1"</span>, <span class="st">"sample2"</span><span class="op">)</span></span></code></pre></div>
<p>In this example, we will use built-in data, which have been converted
into <code>GRanges</code> object previously (<code>CnT_H3K27ac</code>
and <code>CnR_H3K27ac</code>).</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">peaklist</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="va">CnT_H3K27ac</span>, <span class="va">CnR_H3K27ac</span><span class="op">)</span> <span class="co"># create list of peakfiles </span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">peaklist</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"CnT"</span>, <span class="st">"CnR"</span><span class="op">)</span> <span class="co"># set names </span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="blacklist">Blacklist<a class="anchor" aria-label="anchor" href="#blacklist"></a>
</h4>
<p>ENCODE blacklist contains regions of the genome that have anomalous
and/or unstructured signals independent of the cell-line or experiment.
Removal of these regions from peakfiles is recommended for quality
measure.</p>
<p><em>EpiCompare</em> has three built-in blacklist files, hg19, hg38
and mm10, downloaded from ENCODE. Run <code><a href="../reference/hg19_blacklist.html">?hg19_blacklist</a></code> for
more information.</p>
<p>In this example analysis, since all peakfiles
(<code>encode_H3K27ac</code>, <code>CnT_H3K27ac</code>,
<code>CnR_H3K27ac</code>) were generated using human genome reference
build hg19, <code>hg19_blacklist</code> will be used. For hg38, use
<code>data(hg38_blacklist)</code>.</p>
<p>Please ensure that you specify the correct blacklist.</p>
</div>
<div class="section level4">
<h4 id="picard-summary-files">Picard summary files<a class="anchor" aria-label="anchor" href="#picard-summary-files"></a>
</h4>
<p>Note that this is <em>OPTIONAL</em>. If you want the report to
include metrics on DNA fragments (e.g. mapped fragments and duplication
rate), please input summary files from Picard.</p>
<p><a href="https://broadinstitute.github.io/picard/command-line-overview.html#MarkDuplicates" class="external-link">Picard
<em>MarkDuplicates</em></a> can be used to mark duplicate reads that are
found within the alignment. This tool outputs a metrics file with the
ending <code>.MarkDuplicates.metrics.txt</code>. To import this text
file into R as data frame, use:</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">picard</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.table</a></span><span class="op">(</span><span class="st">"/path/to/Picard/.MarkDuplicates.metrics.txt"</span>, header <span class="op">=</span> <span class="cn">TRUE</span>, fill <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<p>In this example. we will use built-in data, which have been converted
into data frame previously (<code>CnT_H3K27ac_picard</code> and
<code>CnR_H3K27ac_picard</code>). The files must be listed and
named:</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># create list of Picard summary</span></span>
<span><span class="va">picard</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="va">CnT_H3K27ac_picard</span>, <span class="va">CnR_H3K27ac_picard</span><span class="op">)</span> </span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">picard</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"CnT"</span>, <span class="st">"CnR"</span><span class="op">)</span> <span class="co"># set names </span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="reference-file">Reference file<a class="anchor" aria-label="anchor" href="#reference-file"></a>
</h4>
<p>This is OPTIONAL. If reference peak file is provided,
<code>stat_plot</code> and <code>chromHMM_plot</code> of overlapping
peaks are included in the report (see <em>Optional plots</em> section
below).</p>
<p>Reference file must be listed and named. In this example, we will use
built-in data (<code>encode_H3K27ac</code>), which has been converted to
<code>GRanges</code> previously:</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">reference_peak</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="st">"ENCODE_H3K27ac"</span> <span class="op">=</span> <span class="va">encode_H3K27ac</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="output-directory">Output Directory<a class="anchor" aria-label="anchor" href="#output-directory"></a>
</h4>
<p>When running <code><a href="../reference/EpiCompare.html">EpiCompare()</a></code>, please ensure that you
specify <code>output_dir</code>. All outputs (figures and HTML report)
will be saved in the specified <code>output_dir</code>.</p>
</div>
</div>
<div class="section level3">
<h3 id="run-epicompare">Run EpiCompare<a class="anchor" aria-label="anchor" href="#run-epicompare"></a>
</h3>
<p>Running EpiCompare is done using the function,
<code><a href="../reference/EpiCompare.html">EpiCompare()</a></code> . Users can choose which analyses to run and
include in the report by setting parameters to <code>TRUE</code> or
<code>FALSE</code>.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/EpiCompare.html">EpiCompare</a></span><span class="op">(</span>peakfiles <span class="op">=</span> <span class="va">peaklist</span>,</span>
<span>           genome_build <span class="op">=</span> <span class="st">"hg19"</span>,</span>
<span>           blacklist <span class="op">=</span> <span class="va">hg19_blacklist</span>,</span>
<span>           picard_files <span class="op">=</span> <span class="va">picard</span>,</span>
<span>           reference <span class="op">=</span> <span class="va">reference_peak</span>,</span>
<span>           upset_plot <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>           stat_plot <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>           chromHMM_plot <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>           chromHMM_annotation <span class="op">=</span> <span class="st">"K562"</span>,</span>
<span>           chipseeker_plot <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>           enrichment_plot <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>           tss_plot <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>           interact <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>           save_output <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>           output_filename <span class="op">=</span> <span class="st">"EpiCompare_test"</span>,</span>
<span>           output_timestamp <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>           output_dir <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## NOTE: The following EpiCompare features are NOT being used: </span></span>
<span><span class="co">##  - upset_plot=</span></span>
<span><span class="co">##  - stat_plot=</span></span>
<span><span class="co">##  - chromHMM_plot=</span></span>
<span><span class="co">##  - chipseeker_plot=</span></span>
<span><span class="co">##  - enrichment_plot=</span></span>
<span><span class="co">##  - tss_plot=</span></span>
<span><span class="co">##  - precision_recall_plot=</span></span>
<span><span class="co">##  - corr_plot=</span></span>
<span><span class="co">##  - add_download_button=</span></span></code></pre>
<pre><code><span><span class="co">## processing file: EpiCompare.Rmd</span></span></code></pre>
<pre><code><span><span class="co">## output file: EpiCompare.knit.md</span></span></code></pre>
<pre><code><span><span class="co">## /usr/bin/pandoc +RTS -K512m -RTS EpiCompare.knit.md --to html4 --from markdown+autolink_bare_uris+tex_math_single_backslash --output /tmp/RtmpbQC8j9/EpiCompare_test.html --lua-filter /__w/_temp/Library/rmarkdown/rmarkdown/lua/pagebreak.lua --lua-filter /__w/_temp/Library/rmarkdown/rmarkdown/lua/latex-div.lua --lua-filter /__w/_temp/Library/rmarkdown/rmarkdown/lua/table-classes.lua --embed-resources --standalone --variable bs3=TRUE --section-divs --table-of-contents --toc-depth 3 --variable toc_float=1 --variable toc_selectors=h1,h2,h3 --variable toc_collapsed=1 --variable toc_smooth_scroll=1 --variable toc_print=1 --template /__w/_temp/Library/rmarkdown/rmd/h/default.html --no-highlight --variable highlightjs=1 --number-sections --variable theme=bootstrap --css custom.css --mathjax --variable 'mathjax-url=https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML' --include-in-header /tmp/RtmpbQC8j9/rmarkdown-str502f451d10c7.html --variable code_folding=hide --variable code_menu=1</span></span></code></pre>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## Output created: /tmp/RtmpbQC8j9/EpiCompare_test.html</span></span></code></pre>
<pre><code><span><span class="co">## [1] "Done in 0.11 min."</span></span></code></pre>
<pre><code><span><span class="co">## All outputs saved to: /tmp/RtmpbQC8j9</span></span></code></pre>
<pre><code><span><span class="co">## [1] "/tmp/RtmpbQC8j9/EpiCompare_test.html"</span></span></code></pre>
<div class="section level4">
<h4 id="optional-plots">Optional plots<a class="anchor" aria-label="anchor" href="#optional-plots"></a>
</h4>
<p>By default, these plots will not be included in the report unless set
<code>TRUE</code>.</p>
<ul>
<li>
<code>upset_plot</code> : Upset plot showing the number of
overlapping peaks between samples. <em>EpiCompare</em> uses
<code>UpSetR</code> package.<br>
</li>
<li>
<code>stat_plot</code> : A <code>reference</code> peakfile must be
included for this plot. The plot displays distribution of statistical
significance (q-values) of sample peaks that are
overlapping/non-overlapping with the <code>reference</code>
dataset.</li>
<li>
<code>chromHMM_plot</code> : <a href="http://compbio.mit.edu/ChromHMM/#:~:text=ChromHMM%20is%20software%20for%20learning,and%20spatial%20patterns%20of%20marks." class="external-link">ChromHMM</a>
annotation of peaks. If <code>reference</code> is provided, ChromHMM
annotation of overlapping and non-overlapping peaks with the
<code>reference</code> is also included in the report.</li>
<li>
<code>chipseeker_plot</code> : <code>ChIPseeker</code> functional
annotation of peaks.</li>
<li>
<code>enrichment_plot</code> : KEGG pathway and GO enrichment
analysis of peaks.</li>
<li>
<code>tss_plot</code> : Peak frequency around (+/- 3000bp)
transcriptional start site. Note that it may take awhile to generate
this plot for large sample sizes.</li>
</ul>
</div>
<div class="section level4">
<h4 id="other-options">Other options<a class="anchor" aria-label="anchor" href="#other-options"></a>
</h4>
<ul>
<li>
<code>chromHMM_annotation</code> : Cell-line annotation for
ChromHMM. Default is K562. Options are:
<ul>
<li>“K562” = K-562 cells</li>
<li>“Gm12878” = Cellosaurus cell-line GM12878</li>
<li>“H1hesc” = H1 Human Embryonic Stem Cell</li>
<li>“Hepg2” = Hep G2 cell</li>
<li>“Hmec” = Human Mammary Epithelial Cell</li>
<li>“Hsmm” = Human Skeletal Muscle Myoblasts</li>
<li>“Huvec” = Human Umbilical Vein Endothelial Cells</li>
<li>“Nhek” = Normal Human Epidermal Keratinocytes</li>
<li>“Nhlf” = Normal Human Lung Fibroblasts</li>
</ul>
</li>
<li>
<code>interact</code> : By default, all heatmaps (percentage overlap
and ChromHMM heatmaps) in the report will be interactive. If set FALSE,
all heatmaps will be static. N.B. If <code>interact=TRUE</code>,
interactive heatmaps will be saved as html files, which may take time
for larger sample sizes.</li>
<li>
<code>output_filename</code> : By default, the report is named
EpiCompare.html. You can specify the filename of the report here.</li>
<li>
<code>output_timestamp</code> : By default FALSE. If TRUE, the
filename of the report includes the date.</li>
</ul>
</div>
</div>
<div class="section level3">
<h3 id="output">Output<a class="anchor" aria-label="anchor" href="#output"></a>
</h3>
<p><em>EpiCompare</em> outputs</p>
<ul>
<li>An HTML report consisting of three sections:
<ul>
<li><ol style="list-style-type: decimal">
<li>
<strong>General Metrics</strong>: Metrics on peaks (percentage of
blacklisted and non-standard peaks, and peak widths) and fragments
(duplication rate) of samples.</li>
</ol></li>
<li><ol start="2" style="list-style-type: decimal">
<li>
<strong>Peak Overlap</strong>: Percentage and statistical
significance of overlapping and non-overlapping peaks. Also includes
upset plot.</li>
</ol></li>
<li><ol start="3" style="list-style-type: decimal">
<li>
<strong>Functional Annotation</strong>: Functional annotation
(ChromHMM, ChIPseeker and enrichment analysis) of peaks. Also includes
peak enrichment around TSS.</li>
</ol></li>
</ul>
</li>
<li>
<code>EpiCompare_file</code> containing all plots generated by
EpiCompare if <code>save_output = TRUE</code>.</li>
</ul>
<p>Both outputs are saved in the specified <code>output_dir</code>.</p>
</div>
</div>
<div class="section level2">
<h2 id="future-enhancements">Future Enhancements<a class="anchor" aria-label="anchor" href="#future-enhancements"></a>
</h2>
<p>In the current version, <em>EpiCompare</em> only recognizes certain
BED formats. We hope to improve this. Moreover, if there are other
downstream analysis tools that may be suitable in <em>EpiCompare</em>,
feel free to report this through <a href="https://github.com/neurogenomics/EpiCompare" class="external-link">Github</a>.</p>
</div>
<div class="section level2">
<h2 id="code-used-to-generate-the-example-report">Code used to generate the example report<a class="anchor" aria-label="anchor" href="#code-used-to-generate-the-example-report"></a>
</h2>
<p>An example report comparing ATAC-seq, DNase-seq and ChIP-seq of K562
can be found <a href="https://neurogenomics.github.io/EpiCompare/articles/example_report.html" class="external-link">here</a>.</p>
<p>Code used to generate this:</p>
<details><div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Load data</span></span>
<span><span class="co"># ATAC-seq data: https://www.encodeproject.org/files/ENCFF558BLC/ </span></span>
<span><span class="va">atac1_hg38</span> <span class="op">&lt;-</span> <span class="fu">ChIPseeker</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ChIPseeker/man/readPeakFile.html" class="external-link">readPeakFile</a></span><span class="op">(</span><span class="st">"/path/to/bed/ENCFF558BLC.bed"</span>, as<span class="op">=</span><span class="st">"GRanges"</span><span class="op">)</span></span>
<span><span class="co"># Dnase-seq data: https://www.encodeproject.org/files/ENCFF274YGF/</span></span>
<span><span class="va">dna1_hg38</span> <span class="op">&lt;-</span> <span class="fu">ChIPseeker</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ChIPseeker/man/readPeakFile.html" class="external-link">readPeakFile</a></span><span class="op">(</span><span class="st">"/path/to/bed/ENCFF274YGF.bed"</span>, as<span class="op">=</span><span class="st">"GRanges"</span><span class="op">)</span></span>
<span><span class="co"># Dnase-seq data: https://www.encodeproject.org/files/ENCFF185XRG/</span></span>
<span><span class="va">dna2_hg38</span> <span class="op">&lt;-</span> <span class="fu">ChIPseeker</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ChIPseeker/man/readPeakFile.html" class="external-link">readPeakFile</a></span><span class="op">(</span><span class="st">"/path/to/bed/ENCFF185XRG.bed"</span>, as<span class="op">=</span><span class="st">"GRanges"</span><span class="op">)</span></span>
<span><span class="co"># ChIP-seq data: https://www.encodeproject.org/files/ENCFF038DDS/</span></span>
<span><span class="va">chip_hg38</span> <span class="op">&lt;-</span> <span class="fu">ChIPseeker</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ChIPseeker/man/readPeakFile.html" class="external-link">readPeakFile</a></span><span class="op">(</span><span class="st">"/path/to/bed/ENCODE_H3K27ac_hg38_ENCFF038DDS.bed"</span>, as<span class="op">=</span><span class="st">"GRanges"</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Peaklist</span></span>
<span><span class="va">peaklist</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="st">"ATAC_ENCFF558BLC"</span> <span class="op">=</span> <span class="va">atac1_hg38_unique</span>,</span>
<span>                 <span class="st">"Dnase_ENCFF274YGF"</span> <span class="op">=</span> <span class="va">dna1_hg38</span>,</span>
<span>                 <span class="st">"ChIP_ENCFF038DDS"</span> <span class="op">=</span> <span class="va">chip_hg38</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Reference</span></span>
<span><span class="va">reference</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="st">"Dnase_ENCFF185XRG_reference"</span><span class="op">=</span><span class="va">dna2_hg38</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Blacklist</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"hg38_blacklist"</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Run Epicompare</span></span>
<span><span class="fu"><a href="../reference/EpiCompare.html">EpiCompare</a></span><span class="op">(</span>peakfiles <span class="op">=</span> <span class="va">peaklist</span>,</span>
<span>           genome_build <span class="op">=</span> <span class="st">"hg38"</span>,</span>
<span>           genome_build_output <span class="op">=</span> <span class="st">"hg38"</span>,</span>
<span>           blacklist <span class="op">=</span> <span class="va">hg38_blacklist</span>,</span>
<span>           reference <span class="op">=</span> <span class="va">reference</span>,</span>
<span>           picard_files <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>           upset_plot <span class="op">=</span> <span class="cn">T</span>,</span>
<span>           stat_plot <span class="op">=</span> <span class="cn">T</span>,</span>
<span>           save_output <span class="op">=</span> <span class="cn">F</span>,</span>
<span>           chromHMM_plot <span class="op">=</span> <span class="cn">T</span>,</span>
<span>           chromHMM_annotation <span class="op">=</span> <span class="st">"K562"</span>,</span>
<span>           chipseeker_plot <span class="op">=</span> <span class="cn">T</span>,</span>
<span>           enrichment_plot <span class="op">=</span> <span class="cn">T</span>,</span>
<span>           tss_plot <span class="op">=</span> <span class="cn">T</span>,</span>
<span>           precision_recall_plot <span class="op">=</span><span class="cn">T</span>,</span>
<span>           corr_plot <span class="op">=</span> <span class="cn">T</span>,</span>
<span>           interact <span class="op">=</span> <span class="cn">T</span>,</span>
<span>           output_dir <span class="op">=</span> <span class="st">"/path/"</span><span class="op">)</span></span></code></pre></div>
</details>
</div>
<div class="section level2">
<h2 id="session-information">Session Information<a class="anchor" aria-label="anchor" href="#session-information"></a>
</h2>
<details><div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">utils</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## R version 4.5.1 (2025-06-13)</span></span>
<span><span class="co">## Platform: x86_64-pc-linux-gnu</span></span>
<span><span class="co">## Running under: Ubuntu 24.04.3 LTS</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Matrix products: default</span></span>
<span><span class="co">## BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3 </span></span>
<span><span class="co">## LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.26.so;  LAPACK version 3.12.0</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## locale:</span></span>
<span><span class="co">##  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              </span></span>
<span><span class="co">##  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    </span></span>
<span><span class="co">##  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   </span></span>
<span><span class="co">##  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 </span></span>
<span><span class="co">##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            </span></span>
<span><span class="co">## [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## time zone: UTC</span></span>
<span><span class="co">## tzcode source: system (glibc)</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## attached base packages:</span></span>
<span><span class="co">## [1] stats     graphics  grDevices utils     datasets  methods   base     </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## other attached packages:</span></span>
<span><span class="co">## [1] EpiCompare_1.13.2 BiocStyle_2.37.1 </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## loaded via a namespace (and not attached):</span></span>
<span><span class="co">##   [1] splines_4.5.1                          </span></span>
<span><span class="co">##   [2] BiocIO_1.19.0                          </span></span>
<span><span class="co">##   [3] bitops_1.0-9                           </span></span>
<span><span class="co">##   [4] ggplotify_0.1.3                        </span></span>
<span><span class="co">##   [5] filelock_1.0.3                         </span></span>
<span><span class="co">##   [6] tibble_3.3.0                           </span></span>
<span><span class="co">##   [7] R.oo_1.27.1                            </span></span>
<span><span class="co">##   [8] XML_3.99-0.19                          </span></span>
<span><span class="co">##   [9] lifecycle_1.0.4                        </span></span>
<span><span class="co">##  [10] httr2_1.2.1                            </span></span>
<span><span class="co">##  [11] lattice_0.22-7                         </span></span>
<span><span class="co">##  [12] magrittr_2.0.4                         </span></span>
<span><span class="co">##  [13] plotly_4.11.0                          </span></span>
<span><span class="co">##  [14] sass_0.4.10                            </span></span>
<span><span class="co">##  [15] rmarkdown_2.29                         </span></span>
<span><span class="co">##  [16] jquerylib_0.1.4                        </span></span>
<span><span class="co">##  [17] yaml_2.3.10                            </span></span>
<span><span class="co">##  [18] plotrix_3.8-4                          </span></span>
<span><span class="co">##  [19] ggtangle_0.0.7                         </span></span>
<span><span class="co">##  [20] cowplot_1.2.0                          </span></span>
<span><span class="co">##  [21] DBI_1.2.3                              </span></span>
<span><span class="co">##  [22] RColorBrewer_1.1-3                     </span></span>
<span><span class="co">##  [23] lubridate_1.9.4                        </span></span>
<span><span class="co">##  [24] abind_1.4-8                            </span></span>
<span><span class="co">##  [25] GenomicRanges_1.61.4                   </span></span>
<span><span class="co">##  [26] purrr_1.1.0                            </span></span>
<span><span class="co">##  [27] R.utils_2.13.0                         </span></span>
<span><span class="co">##  [28] BiocGenerics_0.55.1                    </span></span>
<span><span class="co">##  [29] RCurl_1.98-1.17                        </span></span>
<span><span class="co">##  [30] yulab.utils_0.2.1                      </span></span>
<span><span class="co">##  [31] rappdirs_0.3.3                         </span></span>
<span><span class="co">##  [32] IRanges_2.43.2                         </span></span>
<span><span class="co">##  [33] S4Vectors_0.47.2                       </span></span>
<span><span class="co">##  [34] enrichplot_1.29.2                      </span></span>
<span><span class="co">##  [35] ggrepel_0.9.6                          </span></span>
<span><span class="co">##  [36] tidytree_0.4.6                         </span></span>
<span><span class="co">##  [37] pkgdown_2.1.3                          </span></span>
<span><span class="co">##  [38] ChIPseeker_1.45.0                      </span></span>
<span><span class="co">##  [39] codetools_0.2-20                       </span></span>
<span><span class="co">##  [40] DelayedArray_0.35.3                    </span></span>
<span><span class="co">##  [41] DOSE_4.3.0                             </span></span>
<span><span class="co">##  [42] tidyselect_1.2.1                       </span></span>
<span><span class="co">##  [43] aplot_0.2.9                            </span></span>
<span><span class="co">##  [44] UCSC.utils_1.5.0                       </span></span>
<span><span class="co">##  [45] farver_2.1.2                           </span></span>
<span><span class="co">##  [46] matrixStats_1.5.0                      </span></span>
<span><span class="co">##  [47] stats4_4.5.1                           </span></span>
<span><span class="co">##  [48] BiocFileCache_2.99.6                   </span></span>
<span><span class="co">##  [49] base64enc_0.1-3                        </span></span>
<span><span class="co">##  [50] Seqinfo_0.99.2                         </span></span>
<span><span class="co">##  [51] GenomicAlignments_1.45.4               </span></span>
<span><span class="co">##  [52] jsonlite_2.0.0                         </span></span>
<span><span class="co">##  [53] systemfonts_1.2.3                      </span></span>
<span><span class="co">##  [54] tools_4.5.1                            </span></span>
<span><span class="co">##  [55] treeio_1.33.0                          </span></span>
<span><span class="co">##  [56] TxDb.Hsapiens.UCSC.hg19.knownGene_3.2.2</span></span>
<span><span class="co">##  [57] ragg_1.5.0                             </span></span>
<span><span class="co">##  [58] Rcpp_1.1.0                             </span></span>
<span><span class="co">##  [59] glue_1.8.0                             </span></span>
<span><span class="co">##  [60] SparseArray_1.9.1                      </span></span>
<span><span class="co">##  [61] xfun_0.53                              </span></span>
<span><span class="co">##  [62] qvalue_2.41.0                          </span></span>
<span><span class="co">##  [63] MatrixGenerics_1.21.0                  </span></span>
<span><span class="co">##  [64] GenomeInfoDb_1.45.11                   </span></span>
<span><span class="co">##  [65] dplyr_1.1.4                            </span></span>
<span><span class="co">##  [66] withr_3.0.2                            </span></span>
<span><span class="co">##  [67] BiocManager_1.30.26                    </span></span>
<span><span class="co">##  [68] fastmap_1.2.0                          </span></span>
<span><span class="co">##  [69] boot_1.3-32                            </span></span>
<span><span class="co">##  [70] caTools_1.18.3                         </span></span>
<span><span class="co">##  [71] digest_0.6.37                          </span></span>
<span><span class="co">##  [72] timechange_0.3.0                       </span></span>
<span><span class="co">##  [73] R6_2.6.1                               </span></span>
<span><span class="co">##  [74] mime_0.13                              </span></span>
<span><span class="co">##  [75] gridGraphics_0.5-1                     </span></span>
<span><span class="co">##  [76] seqPattern_1.41.0                      </span></span>
<span><span class="co">##  [77] textshaping_1.0.3                      </span></span>
<span><span class="co">##  [78] GO.db_3.21.0                           </span></span>
<span><span class="co">##  [79] gtools_3.9.5                           </span></span>
<span><span class="co">##  [80] RSQLite_2.4.3                          </span></span>
<span><span class="co">##  [81] R.methodsS3_1.8.2                      </span></span>
<span><span class="co">##  [82] tidyr_1.3.1                            </span></span>
<span><span class="co">##  [83] generics_0.1.4                         </span></span>
<span><span class="co">##  [84] data.table_1.17.8                      </span></span>
<span><span class="co">##  [85] rtracklayer_1.69.1                     </span></span>
<span><span class="co">##  [86] bsplus_0.1.5                           </span></span>
<span><span class="co">##  [87] httr_1.4.7                             </span></span>
<span><span class="co">##  [88] htmlwidgets_1.6.4                      </span></span>
<span><span class="co">##  [89] S4Arrays_1.9.1                         </span></span>
<span><span class="co">##  [90] downloadthis_0.5.0                     </span></span>
<span><span class="co">##  [91] pkgconfig_2.0.3                        </span></span>
<span><span class="co">##  [92] gtable_0.3.6                           </span></span>
<span><span class="co">##  [93] blob_1.2.4                             </span></span>
<span><span class="co">##  [94] S7_0.2.0                               </span></span>
<span><span class="co">##  [95] impute_1.83.0                          </span></span>
<span><span class="co">##  [96] XVector_0.49.1                         </span></span>
<span><span class="co">##  [97] htmltools_0.5.8.1                      </span></span>
<span><span class="co">##  [98] bookdown_0.44                          </span></span>
<span><span class="co">##  [99] fgsea_1.35.6                           </span></span>
<span><span class="co">## [100] scales_1.4.0                           </span></span>
<span><span class="co">## [101] Biobase_2.69.1                         </span></span>
<span><span class="co">## [102] png_0.1-8                              </span></span>
<span><span class="co">## [103] ggfun_0.2.0                            </span></span>
<span><span class="co">## [104] knitr_1.50                             </span></span>
<span><span class="co">## [105] tzdb_0.5.0                             </span></span>
<span><span class="co">## [106] reshape2_1.4.4                         </span></span>
<span><span class="co">## [107] rjson_0.2.23                           </span></span>
<span><span class="co">## [108] uuid_1.2-1                             </span></span>
<span><span class="co">## [109] nlme_3.1-168                           </span></span>
<span><span class="co">## [110] curl_7.0.0                             </span></span>
<span><span class="co">## [111] cachem_1.1.0                           </span></span>
<span><span class="co">## [112] stringr_1.5.2                          </span></span>
<span><span class="co">## [113] BiocVersion_3.22.0                     </span></span>
<span><span class="co">## [114] KernSmooth_2.23-26                     </span></span>
<span><span class="co">## [115] parallel_4.5.1                         </span></span>
<span><span class="co">## [116] AnnotationDbi_1.71.1                   </span></span>
<span><span class="co">## [117] restfulr_0.0.16                        </span></span>
<span><span class="co">## [118] desc_1.4.3                             </span></span>
<span><span class="co">## [119] pillar_1.11.1                          </span></span>
<span><span class="co">## [120] grid_4.5.1                             </span></span>
<span><span class="co">## [121] vctrs_0.6.5                            </span></span>
<span><span class="co">## [122] gplots_3.2.0                           </span></span>
<span><span class="co">## [123] dbplyr_2.5.1                           </span></span>
<span><span class="co">## [124] evaluate_1.0.5                         </span></span>
<span><span class="co">## [125] readr_2.1.5                            </span></span>
<span><span class="co">## [126] GenomicFeatures_1.61.6                 </span></span>
<span><span class="co">## [127] cli_3.6.5                              </span></span>
<span><span class="co">## [128] compiler_4.5.1                         </span></span>
<span><span class="co">## [129] Rsamtools_2.25.3                       </span></span>
<span><span class="co">## [130] rlang_1.1.6                            </span></span>
<span><span class="co">## [131] crayon_1.5.3                           </span></span>
<span><span class="co">## [132] labeling_0.4.3                         </span></span>
<span><span class="co">## [133] plyr_1.8.9                             </span></span>
<span><span class="co">## [134] fs_1.6.6                               </span></span>
<span><span class="co">## [135] ggiraph_0.9.1                          </span></span>
<span><span class="co">## [136] stringi_1.8.7                          </span></span>
<span><span class="co">## [137] genomation_1.41.1                      </span></span>
<span><span class="co">## [138] viridisLite_0.4.2                      </span></span>
<span><span class="co">## [139] gridBase_0.4-7                         </span></span>
<span><span class="co">## [140] BiocParallel_1.43.4                    </span></span>
<span><span class="co">## [141] Biostrings_2.77.2                      </span></span>
<span><span class="co">## [142] lazyeval_0.2.2                         </span></span>
<span><span class="co">## [143] GOSemSim_2.35.1                        </span></span>
<span><span class="co">## [144] Matrix_1.7-4                           </span></span>
<span><span class="co">## [145] BSgenome_1.77.2                        </span></span>
<span><span class="co">## [146] hms_1.1.3                              </span></span>
<span><span class="co">## [147] patchwork_1.3.2                        </span></span>
<span><span class="co">## [148] bit64_4.6.0-1                          </span></span>
<span><span class="co">## [149] ggplot2_4.0.0                          </span></span>
<span><span class="co">## [150] KEGGREST_1.49.1                        </span></span>
<span><span class="co">## [151] SummarizedExperiment_1.39.2            </span></span>
<span><span class="co">## [152] AnnotationHub_3.99.6                   </span></span>
<span><span class="co">## [153] igraph_2.1.4                           </span></span>
<span><span class="co">## [154] memoise_2.0.1                          </span></span>
<span><span class="co">## [155] bslib_0.9.0                            </span></span>
<span><span class="co">## [156] ggtree_3.99.0                          </span></span>
<span><span class="co">## [157] fastmatch_1.1-6                        </span></span>
<span><span class="co">## [158] bit_4.6.0                              </span></span>
<span><span class="co">## [159] ape_5.8-1</span></span></code></pre>
</details>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

      </div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Sera Choi, Brian Schilder, Leyla Abbasova, Alan Murphy, Nathan Skene, Hiranyamaya Dash.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.3.</p>
</div>

      </footer>
</div>






  </body>
</html>
